--- a	2023-09-28 18:47:59.601614922 +0100
+++ b	2023-09-28 18:47:59.629615927 +0100
@@ -74,7 +74,7 @@
 diff --git a/Makefile b/Makefile
 --- a/Makefile
 +++ b/Makefile
-@@ -1425,8 +1425,8 @@ ocamlprof_LIBRARIES =
+@@ -1426,8 +1426,8 @@ ocamlprof_LIBRARIES =
  ocamlprof_MODULES = \
    config build_path_prefix_map misc identifiable numbers arg_helper \
    local_store load_path clflags terminfo warnings location longident \
