--- a	2023-09-28 18:44:22.781372700 +0100
+++ b	2023-09-28 18:44:22.789373024 +0100
@@ -1,133 +1,155 @@
 diff --git a/testsuite/tests/parallel/atomics.ml b/testsuite/tests/parallel/atomics.ml
 --- a/testsuite/tests/parallel/atomics.ml
 +++ b/testsuite/tests/parallel/atomics.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  type u = U of unit
 diff --git a/testsuite/tests/parallel/constpromote.ml b/testsuite/tests/parallel/constpromote.ml
 --- a/testsuite/tests/parallel/constpromote.ml
 +++ b/testsuite/tests/parallel/constpromote.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  (* when run with the bytecode debug runtime, this test
 diff --git a/testsuite/tests/parallel/deadcont.ml b/testsuite/tests/parallel/deadcont.ml
 --- a/testsuite/tests/parallel/deadcont.ml
 +++ b/testsuite/tests/parallel/deadcont.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  (*
 diff --git a/testsuite/tests/parallel/domain_dls.ml b/testsuite/tests/parallel/domain_dls.ml
 --- a/testsuite/tests/parallel/domain_dls.ml
 +++ b/testsuite/tests/parallel/domain_dls.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  let check_dls () =
 diff --git a/testsuite/tests/parallel/domain_dls2.ml b/testsuite/tests/parallel/domain_dls2.ml
 --- a/testsuite/tests/parallel/domain_dls2.ml
 +++ b/testsuite/tests/parallel/domain_dls2.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  let _ =
 diff --git a/testsuite/tests/parallel/domain_id.ml b/testsuite/tests/parallel/domain_id.ml
 --- a/testsuite/tests/parallel/domain_id.ml
 +++ b/testsuite/tests/parallel/domain_id.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  open Domain
 diff --git a/testsuite/tests/parallel/domain_parallel_spawn_burn.ml b/testsuite/tests/parallel/domain_parallel_spawn_burn.ml
 --- a/testsuite/tests/parallel/domain_parallel_spawn_burn.ml
 +++ b/testsuite/tests/parallel/domain_parallel_spawn_burn.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  open Domain
 diff --git a/testsuite/tests/parallel/domain_parallel_spawn_burn_gc_set.ml b/testsuite/tests/parallel/domain_parallel_spawn_burn_gc_set.ml
 --- a/testsuite/tests/parallel/domain_parallel_spawn_burn_gc_set.ml
 +++ b/testsuite/tests/parallel/domain_parallel_spawn_burn_gc_set.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  open Domain
 diff --git a/testsuite/tests/parallel/join.ml b/testsuite/tests/parallel/join.ml
 --- a/testsuite/tests/parallel/join.ml
 +++ b/testsuite/tests/parallel/join.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  let test_size =
 diff --git a/testsuite/tests/parallel/pingpong.ml b/testsuite/tests/parallel/pingpong.ml
 --- a/testsuite/tests/parallel/pingpong.ml
 +++ b/testsuite/tests/parallel/pingpong.ml
-@@ -1,8 +1,4 @@
+@@ -1,10 +1,6 @@
  (* TEST
--* hasunix
+ * skip
+ reason = "OCaml 5 only"
+-** hasunix
 -include unix
--** bytecode
--** native
+-*** bytecode
+-*** native
  *)
  
  let r = ref (Some 0)
 diff --git a/testsuite/tests/parallel/prodcons_domains.ml b/testsuite/tests/parallel/prodcons_domains.ml
 --- a/testsuite/tests/parallel/prodcons_domains.ml
 +++ b/testsuite/tests/parallel/prodcons_domains.ml
-@@ -1,10 +1,4 @@
+@@ -1,12 +1,6 @@
  (* TEST
 -
--* hassysthreads
+ * skip
+ reason = "OCaml 5 only"
+-** hassysthreads
 -include systhreads
--** bytecode
--** native
+-*** bytecode
+-*** native
 -
  *)
  
