--- a	2023-09-28 18:45:38.892380650 +0100
+++ b	2023-09-28 18:45:38.908381268 +0100
@@ -1,7 +1,7 @@
 diff --git a/.gitattributes b/.gitattributes
 --- a/.gitattributes
 +++ b/.gitattributes
-@@ -192,14 +192,6 @@ tools/pre-commit-githook text eol=lf
+@@ -193,14 +193,6 @@ tools/pre-commit-githook text eol=lf
  tools/markdown-add-pr-links.sh text eol=lf
  runtime/caml/sizeclasses.h typo.missing-header typo.white-at-eol
  
@@ -31,7 +31,7 @@
 diff --git a/Makefile b/Makefile
 --- a/Makefile
 +++ b/Makefile
-@@ -729,7 +729,7 @@ runtime-all: \
+@@ -726,7 +726,7 @@ runtime-all: \
    $(runtime_PROGRAMS) $(SAK)
  
  .PHONY: runtime-allopt
@@ -40,7 +40,7 @@
  runtime-allopt: \
    $(runtime_NATIVE_STATIC_LIBRARIES) $(runtime_NATIVE_SHARED_LIBRARIES)
  else
-@@ -897,7 +897,7 @@ $(DEPDIR)/runtime/%.npic.$(D): \
+@@ -894,7 +894,7 @@ $(DEPDIR)/runtime/%.npic.$(D): \
  # that corresponds to the name of the generated object file
  # (without the extension, which is added by the macro)
  define COMPILE_C_FILE
@@ -49,7 +49,7 @@
  ifneq "$(1)" "%"
  # -MG would ensure that the dependencies are generated even if the files listed
  # in $$(runtime_BUILT_HEADERS) haven't been assembled yet. However,
-@@ -915,7 +915,7 @@ else
+@@ -912,7 +912,7 @@ else
  $(1).$(O): $(2).c \
    $(runtime_CONFIGURED_HEADERS) $(runtime_BUILT_HEADERS) \
    $(RUNTIME_HEADERS)
@@ -58,7 +58,7 @@
  	$$(CC) -c $$(OC_CFLAGS) $$(CFLAGS) $$(OC_CPPFLAGS) $$(CPPFLAGS) \
  	  $$(OUTPUTOBJ)$$@ $$<
  endef
-@@ -924,7 +924,7 @@ $(DEPDIR)/runtime:
+@@ -921,7 +921,7 @@ $(DEPDIR)/runtime:
  	$(MKDIR) $@
  
  runtime_OBJECT_TYPES = % %.b %.bd %.bi %.bpic
@@ -67,7 +67,7 @@
  runtime_OBJECT_TYPES += %.n %.nd %.ni %.np %.npic
  endif
  
-@@ -993,7 +993,7 @@ runtime/%_libasmrunpic.obj: runtime/%.asm
+@@ -990,7 +990,7 @@ runtime/%_libasmrunpic.obj: runtime/%.asm
  
  runtime_DEP_FILES := $(addsuffix .b, \
    $(basename $(runtime_BYTECODE_C_SOURCES) runtime/instrtrace))
@@ -153,7 +153,7 @@
 diff --git a/api_docgen/Makefile.docfiles b/api_docgen/Makefile.docfiles
 --- a/api_docgen/Makefile.docfiles
 +++ b/api_docgen/Makefile.docfiles
-@@ -43,19 +43,19 @@ stdlib_UNPREFIXED=$(STDLIB_MODULE_BASENAMES)
+@@ -44,19 +44,19 @@ stdlib_UNPREFIXED=$(STDLIB_MODULE_BASENAMES)
  
  otherlibref := $(dynlink_MLIS:%.mli=%)
  
@@ -204,7 +204,7 @@
  AC_SUBST([documentation_tool])
  AC_SUBST([documentation_tool_cmd])
  AC_SUBST([ocamltest])
-@@ -491,8 +492,8 @@ AS_IF([test x"$enable_unix_lib" = "xno" || test x"$enable_str_lib" = "xno"],
+@@ -502,8 +503,8 @@ AS_IF([test x"$enable_unix_lib" = "xno" || test x"$enable_str_lib" = "xno"],
    [AS_IF([test x"$enable_ocamldoc" = "xyes"],
      [AC_MSG_ERROR([ocamldoc requires the unix and str libraries])],
      [enable_ocamldoc="no"
@@ -215,15 +215,15 @@
  
  # Initialization of libtool
  # Allow the MSVC linker to be found even if ld isn't installed.
-@@ -619,6 +620,7 @@ AS_CASE([$host],
- 
- otherlibraries="dynlink runtime_events"
+@@ -631,6 +632,7 @@ AS_CASE([$host],
+ dnl otherlibraries="dynlink runtime_events"
+ otherlibraries="dynlink"
  lib_dynlink=true
 +lib_runtime_events=true
  AS_IF([test x"$enable_unix_lib" != "xno"],
    [enable_unix_lib=yes
    AC_CONFIG_FILES([otherlibs/unix/META])
-@@ -1954,14 +1956,14 @@ AC_COMPILE_IFELSE(
+@@ -1966,14 +1968,14 @@ AC_COMPILE_IFELSE(
  AS_CASE([$enable_debugger],
    [no],
      [with_debugger=""
@@ -241,7 +241,7 @@
      AC_MSG_NOTICE([replay debugger not supported])])
    ])
  
-@@ -2041,9 +2043,9 @@ AS_IF([test x"$enable_installing_source_artifacts" = "xno"],
+@@ -2089,9 +2091,9 @@ AS_IF([test x"$enable_installing_source_artifacts" = "xno"],
  
  AS_IF([test x"$enable_ocamldoc" = "xno"],
    [ocamldoc=""
@@ -456,7 +456,7 @@
 diff --git a/tools/Makefile b/tools/Makefile
 --- a/tools/Makefile
 +++ b/tools/Makefile
-@@ -301,13 +301,13 @@ CMPBYT=$(ROOTDIR)/compilerlibs/ocamlcommon.cma \
+@@ -308,13 +308,13 @@ CMPBYT=$(ROOTDIR)/compilerlibs/ocamlcommon.cma \
  cmpbyt$(EXE): $(CMPBYT)
  cmpbyt.opt$(EXE): $(call byte2native, $(CMPBYT))
  
@@ -472,7 +472,7 @@
  
  # checkstack tool
  
-@@ -316,24 +316,24 @@ checkstack$(EXE): checkstack.$(O)
+@@ -323,24 +323,24 @@ checkstack$(EXE): checkstack.$(O)
  
  #Scan latex files, and run ocaml code examples
  
