--- a	2023-09-28 18:44:12.400951348 +0100
+++ b	2023-09-28 18:44:12.404951511 +0100
@@ -1,8 +1,8 @@
 diff --git a/testsuite/tests/weak-ephe-final/weaklifetime_par.ml b/testsuite/tests/weak-ephe-final/weaklifetime_par.ml
 --- a/testsuite/tests/weak-ephe-final/weaklifetime_par.ml
 +++ b/testsuite/tests/weak-ephe-final/weaklifetime_par.ml
-@@ -1,6 +1,8 @@
- (* TEST
+@@ -3,6 +3,8 @@
+    reason = "OCaml 5 only"
  *)
  
 +let () = Random.self_init ()
