--- a	2023-09-28 18:43:24.646975483 +0100
+++ b	2023-09-28 18:43:24.658975988 +0100
@@ -11,50 +11,14 @@
  \subsubsection*{sss:stdlib-misc}{Misc:}
  \begin{tabular}{lll}
 @@ -133,7 +133,7 @@ be called from C \\
- \stddocitem{Condition}{condition variables to synchronize between threads}
- \stddocitem{Domain}{Domain spawn/join and domain local variables}
+ %\stddocitem{Condition}{condition variables to synchronize between threads}
+ %\stddocitem{Domain}{Domain spawn/join and domain local variables}
  \stddocitem{Digest}{MD5 message digest}
--\stddocitem{EffectHandlers}{deep and shallow effect handlers}
-+\stddocitem{Effect}{deep and shallow effect handlers}
+-%\stddocitem{EffectHandlers}{deep and shallow effect handlers}
++%\stddocitem{Effect}{deep and shallow effect handlers}
  \stddocitem{Either}{either values}
  \stddocitem{Ephemeron}{Ephemerons and weak hash table}
  \stddocitem{Filename}{operations on file names}
-diff --git a/stdlib/.depend b/stdlib/.depend
---- a/stdlib/.depend
-+++ b/stdlib/.depend
-@@ -229,15 +229,15 @@ stdlib__Domain.cmx : domain.ml \
-     stdlib__Array.cmx \
-     stdlib__Domain.cmi
- stdlib__Domain.cmi : domain.mli
--stdlib__EffectHandlers.cmo : effectHandlers.ml \
-+stdlib__Effect.cmo : effect.ml \
-     stdlib__Printexc.cmi \
-     stdlib__Obj.cmi \
--    stdlib__EffectHandlers.cmi
--stdlib__EffectHandlers.cmx : effectHandlers.ml \
-+    stdlib__Effect.cmi
-+stdlib__Effect.cmx : effect.ml \
-     stdlib__Printexc.cmx \
-     stdlib__Obj.cmx \
--    stdlib__EffectHandlers.cmi
--stdlib__EffectHandlers.cmi : effectHandlers.mli \
-+    stdlib__Effect.cmi
-+stdlib__Effect.cmi : effect.mli \
-     stdlib__Printexc.cmi
- stdlib__Either.cmo : either.ml \
-     stdlib__Either.cmi
-diff --git a/stdlib/StdlibModules b/stdlib/StdlibModules
---- a/stdlib/StdlibModules
-+++ b/stdlib/StdlibModules
-@@ -45,7 +45,7 @@ STDLIB_MODULE_BASENAMES = \
-   printexc fun gc digest random hashtbl weak \
-   format scanf callback camlinternalOO oo camlinternalMod genlex ephemeron \
-   filename complex arrayLabels listLabels bytesLabels stringLabels moreLabels \
--  stdLabels bigarray in_channel out_channel effectHandlers
-+  stdLabels bigarray in_channel out_channel effect
- 
- STDLIB_PREFIXED_MODULES = \
-   $(filter-out stdlib camlinternal%, $(STDLIB_MODULE_BASENAMES))
 diff --git a/stdlib/effectHandlers.ml b/stdlib/effect.ml
 similarity index 100%
 rename from stdlib/effectHandlers.ml
@@ -66,32 +30,32 @@
 diff --git a/stdlib/stdlib.ml b/stdlib/stdlib.ml
 --- a/stdlib/stdlib.ml
 +++ b/stdlib/stdlib.ml
-@@ -585,7 +585,7 @@ module Complex        = Complex
- module Condition      = Condition
+@@ -595,7 +595,7 @@ module Condition      = Condition
  module Digest         = Digest
+ (* BACKPORT
  module Domain         = Domain
 -module EffectHandlers = EffectHandlers
 +module Effect         = Effect
+ *)
  module Either         = Either
  module Ephemeron      = Ephemeron
- module Filename       = Filename
 diff --git a/stdlib/stdlib.mli b/stdlib/stdlib.mli
 --- a/stdlib/stdlib.mli
 +++ b/stdlib/stdlib.mli
-@@ -1379,7 +1379,7 @@ module Complex        = Complex
- module Condition      = Condition
+@@ -1382,7 +1382,7 @@ module Condition      = Condition
  module Digest         = Digest
+ (* BACKPORT
  module Domain         = Domain
 -module EffectHandlers = EffectHandlers
 +module Effect         = Effect
+ *)
  module Either         = Either
  module Ephemeron      = Ephemeron
- module Filename       = Filename
 diff --git a/testsuite/tests/backtrace/backtrace_effects.ml b/testsuite/tests/backtrace/backtrace_effects.ml
 --- a/testsuite/tests/backtrace/backtrace_effects.ml
 +++ b/testsuite/tests/backtrace/backtrace_effects.ml
-@@ -4,8 +4,8 @@
-    exit_status = "2"
+@@ -6,8 +6,8 @@
+    reason = "OCaml 5 only"
  *)
  
 -open EffectHandlers
@@ -104,8 +68,8 @@
 diff --git a/testsuite/tests/backtrace/backtrace_effects_nested.ml b/testsuite/tests/backtrace/backtrace_effects_nested.ml
 --- a/testsuite/tests/backtrace/backtrace_effects_nested.ml
 +++ b/testsuite/tests/backtrace/backtrace_effects_nested.ml
-@@ -2,8 +2,8 @@
-    flags = "-g"
+@@ -4,8 +4,8 @@
+    reason = "OCaml 5 only"
  *)
  
 -open EffectHandlers
@@ -118,7 +82,7 @@
 diff --git a/testsuite/tests/callback/nested_fiber.ml b/testsuite/tests/callback/nested_fiber.ml
 --- a/testsuite/tests/callback/nested_fiber.ml
 +++ b/testsuite/tests/callback/nested_fiber.ml
-@@ -8,8 +8,8 @@
+@@ -10,8 +10,8 @@
  
  external caml_to_c : (unit -> 'a) -> 'a = "caml_to_c"
  
@@ -132,7 +96,7 @@
 diff --git a/testsuite/tests/callback/stack_overflow.ml b/testsuite/tests/callback/stack_overflow.ml
 --- a/testsuite/tests/callback/stack_overflow.ml
 +++ b/testsuite/tests/callback/stack_overflow.ml
-@@ -14,8 +14,8 @@ let rec deep = function
+@@ -16,8 +16,8 @@ let rec deep = function
    | n ->
       caml_to_c (fun () -> deep (n-1))
  
@@ -146,7 +110,7 @@
 diff --git a/testsuite/tests/callback/test7.ml b/testsuite/tests/callback/test7.ml
 --- a/testsuite/tests/callback/test7.ml
 +++ b/testsuite/tests/callback/test7.ml
-@@ -10,8 +10,8 @@
+@@ -12,8 +12,8 @@
   * OCaml (c_to_caml) to C (printf functions). Effect E is performed in the
   * callback, which does not have a handler. *)
  
@@ -160,8 +124,8 @@
 diff --git a/testsuite/tests/effects/backtrace.ml b/testsuite/tests/effects/backtrace.ml
 --- a/testsuite/tests/effects/backtrace.ml
 +++ b/testsuite/tests/effects/backtrace.ml
-@@ -3,8 +3,8 @@
-    ocamlrunparam += ",b=1"
+@@ -5,8 +5,8 @@
+    reason = "OCaml 5 only"
  *)
  
 -open EffectHandlers
@@ -184,8 +148,8 @@
 diff --git a/testsuite/tests/effects/cmphash.ml b/testsuite/tests/effects/cmphash.ml
 --- a/testsuite/tests/effects/cmphash.ml
 +++ b/testsuite/tests/effects/cmphash.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -198,8 +162,8 @@
 diff --git a/testsuite/tests/effects/evenodd.ml b/testsuite/tests/effects/evenodd.ml
 --- a/testsuite/tests/effects/evenodd.ml
 +++ b/testsuite/tests/effects/evenodd.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -224,7 +188,7 @@
 diff --git a/testsuite/tests/effects/issue479.ml b/testsuite/tests/effects/issue479.ml
 --- a/testsuite/tests/effects/issue479.ml
 +++ b/testsuite/tests/effects/issue479.ml
-@@ -4,8 +4,8 @@
+@@ -6,8 +6,8 @@
  
  (* https://github.com/ocaml-multicore/ocaml-multicore/issues/479 *)
  
@@ -238,8 +202,8 @@
 diff --git a/testsuite/tests/effects/overflow.ml b/testsuite/tests/effects/overflow.ml
 --- a/testsuite/tests/effects/overflow.ml
 +++ b/testsuite/tests/effects/overflow.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -252,8 +216,8 @@
 diff --git a/testsuite/tests/effects/partial.ml b/testsuite/tests/effects/partial.ml
 --- a/testsuite/tests/effects/partial.ml
 +++ b/testsuite/tests/effects/partial.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -266,8 +230,8 @@
 diff --git a/testsuite/tests/effects/reperform.ml b/testsuite/tests/effects/reperform.ml
 --- a/testsuite/tests/effects/reperform.ml
 +++ b/testsuite/tests/effects/reperform.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -280,8 +244,8 @@
 diff --git a/testsuite/tests/effects/sched.ml b/testsuite/tests/effects/sched.ml
 --- a/testsuite/tests/effects/sched.ml
 +++ b/testsuite/tests/effects/sched.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -294,8 +258,8 @@
 diff --git a/testsuite/tests/effects/shallow_state.ml b/testsuite/tests/effects/shallow_state.ml
 --- a/testsuite/tests/effects/shallow_state.ml
 +++ b/testsuite/tests/effects/shallow_state.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -308,8 +272,8 @@
 diff --git a/testsuite/tests/effects/shallow_state_io.ml b/testsuite/tests/effects/shallow_state_io.ml
 --- a/testsuite/tests/effects/shallow_state_io.ml
 +++ b/testsuite/tests/effects/shallow_state_io.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -322,8 +286,8 @@
 diff --git a/testsuite/tests/effects/test1.ml b/testsuite/tests/effects/test1.ml
 --- a/testsuite/tests/effects/test1.ml
 +++ b/testsuite/tests/effects/test1.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -336,8 +300,8 @@
 diff --git a/testsuite/tests/effects/test10.ml b/testsuite/tests/effects/test10.ml
 --- a/testsuite/tests/effects/test10.ml
 +++ b/testsuite/tests/effects/test10.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -350,7 +314,7 @@
 diff --git a/testsuite/tests/effects/test11.ml b/testsuite/tests/effects/test11.ml
 --- a/testsuite/tests/effects/test11.ml
 +++ b/testsuite/tests/effects/test11.ml
-@@ -4,8 +4,8 @@
+@@ -6,8 +6,8 @@
  (* Tests RESUMETERM with extra_args != 0 in bytecode,
     by calling a handler with a tail-continue that returns a function *)
  
@@ -364,7 +328,7 @@
 diff --git a/testsuite/tests/effects/test2.ml b/testsuite/tests/effects/test2.ml
 --- a/testsuite/tests/effects/test2.ml
 +++ b/testsuite/tests/effects/test2.ml
-@@ -2,8 +2,8 @@
+@@ -4,8 +4,8 @@
   *)
  
  open Printf
@@ -378,8 +342,8 @@
 diff --git a/testsuite/tests/effects/test3.ml b/testsuite/tests/effects/test3.ml
 --- a/testsuite/tests/effects/test3.ml
 +++ b/testsuite/tests/effects/test3.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -392,8 +356,8 @@
 diff --git a/testsuite/tests/effects/test4.ml b/testsuite/tests/effects/test4.ml
 --- a/testsuite/tests/effects/test4.ml
 +++ b/testsuite/tests/effects/test4.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -406,8 +370,8 @@
 diff --git a/testsuite/tests/effects/test5.ml b/testsuite/tests/effects/test5.ml
 --- a/testsuite/tests/effects/test5.ml
 +++ b/testsuite/tests/effects/test5.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -420,8 +384,8 @@
 diff --git a/testsuite/tests/effects/test6.ml b/testsuite/tests/effects/test6.ml
 --- a/testsuite/tests/effects/test6.ml
 +++ b/testsuite/tests/effects/test6.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -434,8 +398,9 @@
 diff --git a/testsuite/tests/effects/test_lazy.ml b/testsuite/tests/effects/test_lazy.ml
 --- a/testsuite/tests/effects/test_lazy.ml
 +++ b/testsuite/tests/effects/test_lazy.ml
-@@ -1,7 +1,7 @@
- (* TEST *)
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
+ *)
  
 -open EffectHandlers
 -open EffectHandlers.Deep
@@ -447,8 +412,8 @@
 diff --git a/testsuite/tests/effects/used_cont.ml b/testsuite/tests/effects/used_cont.ml
 --- a/testsuite/tests/effects/used_cont.ml
 +++ b/testsuite/tests/effects/used_cont.ml
-@@ -1,8 +1,8 @@
- (* TEST
+@@ -3,8 +3,8 @@
+    reason = "OCaml 5 only"
   *)
  
 -open EffectHandlers
@@ -461,7 +426,7 @@
 diff --git a/testsuite/tests/parallel/deadcont.ml b/testsuite/tests/parallel/deadcont.ml
 --- a/testsuite/tests/parallel/deadcont.ml
 +++ b/testsuite/tests/parallel/deadcont.ml
-@@ -11,8 +11,8 @@ include unix
+@@ -13,8 +13,8 @@ include unix
      https://github.com/ocamllabs/ocaml-multicore/issues/175
   *)
  
@@ -475,7 +440,7 @@
 diff --git a/testsuite/tests/parallel/mctest.ml b/testsuite/tests/parallel/mctest.ml
 --- a/testsuite/tests/parallel/mctest.ml
 +++ b/testsuite/tests/parallel/mctest.ml
-@@ -150,8 +150,8 @@ end
+@@ -152,8 +152,8 @@ end
  
  module Scheduler =
  struct
