--- a	2023-09-28 18:46:04.621367325 +0100
+++ b	2023-09-28 18:46:04.637367934 +0100
@@ -1,7 +1,7 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -736,6 +736,10 @@ OCaml 5.0
+@@ -731,6 +731,10 @@ OCaml 5.0
    mismatch error involving recursive types.
    (Florian Angeletti, review by Gabriel Scherer)
  
@@ -9,6 +9,6 @@
 +  the bug was only present in development version of OCaml 5.
 +  (Stephen Dolan, report by Andre Maroneze, review by Gabriel Scherer)
 +
- OCaml 4.14.0 (28 March 2022)
- ----------------------------
+ OCaml 4.14 maintenance branch
+ -----------------------------
  
