--- a	2023-09-28 18:44:06.100694236 +0100
+++ b	2023-09-28 18:44:06.108694563 +0100
@@ -26,7 +26,7 @@
  # Note: This is present for the flexdll bootstrap where it exposed as the old
  # TOOLPREF variable. It would be better if flexdll where updated to require
  # WINDRES instead.
-@@ -302,7 +303,7 @@ AC_ARG_WITH([odoc],
+@@ -304,7 +305,7 @@ AC_ARG_WITH([odoc],
  
  AC_ARG_ENABLE([ocamltest],
    [AS_HELP_STRING([--disable-ocamltest],
@@ -35,7 +35,7 @@
  
  AC_ARG_ENABLE([native-toplevel],
    [AS_HELP_STRING([--enable-native-toplevel],
-@@ -1878,6 +1879,22 @@ AS_CASE([$enable_ocamltest,OCAML__DEVELOPMENT_VERSION],
+@@ -1916,6 +1917,22 @@ AS_CASE([$enable_ocamltest,OCAML__DEVELOPMENT_VERSION],
    [yes,*|,true],[ocamltest='ocamltest'],
    [ocamltest=''])
  
@@ -85,7 +85,7 @@
 +	    $(call SUBST,DIFF_FLAGS) \
  	    $(call SUBST,WINDOWS_UNICODE) \
  	    $(call SUBST,FUNCTION_SECTIONS) \
- 	    $< > $@
+ 	    $(call SUBST,NAKED_POINTERS) \
 diff --git a/ocamltest/filecompare.ml b/ocamltest/filecompare.ml
 --- a/ocamltest/filecompare.ml
 +++ b/ocamltest/filecompare.ml
