--- a	2023-09-28 18:48:34.758866630 +0100
+++ b	2023-09-28 18:48:34.770867054 +0100
@@ -165,17 +165,17 @@
  - #11533, #11534: follow synonyms again in #show_module_type
    (this had stopped working in 4.14.0)
    (Gabriel Scherer, review by Jacques Garrigue and Florian Angeletti,
-@@ -1947,9 +1956,6 @@ OCaml 4.14.0 (28 March 2022)
+@@ -1976,9 +1985,6 @@ OCaml 4.14.0 (28 March 2022)
  - #10786: The implementation of Complex.norm now uses Float.hypot.
    (Christophe Troestler, review by David Allsopp and Xavier Leroy)
  
 -- #10464: Add List.is_empty.
 -  (Craig Ferguson, review by David Allsopp)
 -
- ### Other libraries:
+ * #10922: Add deprecation warnings on {Int32,Int64,Nativeint}.format.
+   (Nicolás Ojeda Bär, review by Xavier Leroy and Florian Angeletti)
  
- - #10192: Add support for Unix domain sockets on Windows and use them
-@@ -2021,10 +2027,6 @@ OCaml 4.14.0 (28 March 2022)
+@@ -2053,10 +2059,6 @@ OCaml 4.14.0 (28 March 2022)
  
  ### Manual and documentation:
  
