--- a	2023-09-28 18:46:06.485438257 +0100
+++ b	2023-09-28 18:46:06.489438409 +0100
@@ -1,16 +1,20 @@
 diff --git a/testsuite/tests/lib-runtime-events/test_instrumented.ml b/testsuite/tests/lib-runtime-events/test_instrumented.ml
 --- a/testsuite/tests/lib-runtime-events/test_instrumented.ml
 +++ b/testsuite/tests/lib-runtime-events/test_instrumented.ml
-@@ -1,7 +1,9 @@
+@@ -1,9 +1,11 @@
  (* TEST
--  * native
+-  * skip
+-  reason = "OCaml 5 only"
+-  ** native
 -    include runtime_events
 -    flags = "-runtime-variant=i"
 +   include runtime_events
 +   flags = "-runtime-variant=i"
 +
-+   * instrumented-runtime
-+   ** native
++   * skip
++   reason = "OCaml 5 only"
++   ** instrumented-runtime
++   *** native
  *)
  
  open Runtime_events
