--- a	2023-09-28 18:43:51.732103868 +0100
+++ b	2023-09-28 18:43:51.744104364 +0100
@@ -1,7 +1,7 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -135,12 +135,6 @@ Working version
+@@ -138,12 +138,6 @@ Working version
  
  ### Bug fixes:
  
@@ -14,7 +14,7 @@
  - #11025: Do not pass -no-pie to the C compiler on musl/arm64
    (omni and Kate Deplaix, review by Xavier Leroy)
  
-@@ -625,6 +619,12 @@ OCaml 4.14.0
+@@ -634,6 +628,12 @@ OCaml 4.14.0
  - #10907, #10959: Wrong type inferred from existential types
    (Jacques Garrigue and Gabriel Scherer, report by @dyzsr, review by Leo White)
  
