--- a	2023-09-28 18:43:26.063034996 +0100
+++ b	2023-09-28 18:43:26.067035164 +0100
@@ -12,7 +12,7 @@
  #ifdef POSIX_SIGNALS
  
  static void decode_sigset(value vset, sigset_t * set)
-@@ -80,11 +76,17 @@ CAMLprim value unix_sigprocmask(value vaction, value vset)
+@@ -91,15 +87,17 @@ CAMLprim value unix_sigprocmask(value vaction, value vset)
  CAMLprim value unix_sigpending(value unit)
  {
    sigset_t pending;
@@ -21,7 +21,11 @@
 +  uintnat curr;
    if (sigpending(&pending) == -1) uerror("sigpending", Nothing);
 -  for (i = 1; i < NSIG; i++)
+-#if 0 /* BACKPORT BEGIN */
 -    if(atomic_load_explicit(&caml_pending_signals[i], memory_order_seq_cst))
+-#endif
+-    if(caml_pending_signals[i])
+-/* BACKPORT END */
 -      sigaddset(&pending, i);
 +  for (i = 0; i < NSIG_WORDS; i++) {
 +    curr = atomic_load(&caml_pending_signals[i]);
