--- a	2023-09-28 18:45:34.784221761 +0100
+++ b	2023-09-28 18:45:34.788221916 +0100
@@ -3,7 +3,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -627,7 +627,8 @@ AS_IF([test x"$enable_unix_lib" != "xno"],
+@@ -639,7 +639,8 @@ AS_IF([test x"$enable_unix_lib" != "xno"],
    ocamltest_unix_impl="real"
    ocamltest_unix_name="unix"
    ocamltest_unix_path='$(ROOTDIR)/otherlibs/unix'
