--- a	2023-09-28 18:47:43.649040156 +0100
+++ b	2023-09-28 18:47:43.669040880 +0100
@@ -35,7 +35,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -307,11 +307,17 @@ AC_ARG_ENABLE([debug-runtime],
+@@ -310,11 +310,17 @@ AC_ARG_ENABLE([debug-runtime],
    [AS_HELP_STRING([--disable-debug-runtime],
      [do not build runtime with debugging support])])
  
@@ -56,7 +56,7 @@
  
  AC_ARG_ENABLE([dependency-generation],
    [AS_HELP_STRING([--disable-dependency-generation],
-@@ -489,9 +495,9 @@ AC_ARG_WITH([zstd],
+@@ -500,9 +506,9 @@ AC_ARG_WITH([zstd],
      [disable compression of marshaled data])])
  
  AS_IF([test x"$enable_unix_lib" = "xno"],
@@ -69,7 +69,7 @@
  
  AS_IF([test x"$enable_unix_lib" = "xno" || test x"$enable_str_lib" = "xno"],
    [AS_IF([test x"$enable_ocamldoc" = "xyes"],
-@@ -2003,18 +2009,18 @@ AS_IF([test x"$zstd_status" = "xok"],
+@@ -2019,18 +2025,18 @@ AS_IF([test x"$zstd_status" = "xok"],
  
  ## Determine whether the debugger should/can be built
  
