--- a	2023-09-28 18:44:54.894658841 +0100
+++ b	2023-09-28 18:44:54.918659793 +0100
@@ -6457,7 +6457,7 @@
 diff --git a/stdlib/.depend b/stdlib/.depend
 --- a/stdlib/.depend
 +++ b/stdlib/.depend
-@@ -266,20 +266,13 @@ stdlib__Ephemeron.cmx : ephemeron.ml \
+@@ -229,14 +229,6 @@ stdlib__Ephemeron.cmx : ephemeron.ml \
  stdlib__Ephemeron.cmi : ephemeron.mli \
      stdlib__Seq.cmi \
      stdlib__Hashtbl.cmi
@@ -6472,22 +6472,23 @@
  stdlib__Filename.cmo : filename.ml \
      stdlib__Sys.cmi \
      stdlib__String.cmi \
-     stdlib__Random.cmi \
+@@ -244,6 +236,7 @@ stdlib__Filename.cmo : filename.ml \
      stdlib__Printf.cmi \
      stdlib__List.cmi \
+     stdlib__Lazy.cmi \
 +    stdlib__Fun.cmi \
-     stdlib__Domain.cmi \
      stdlib__Buffer.cmi \
      stdlib__Filename.cmi
-@@ -289,6 +282,7 @@ stdlib__Filename.cmx : filename.ml \
-     stdlib__Random.cmx \
+ stdlib__Filename.cmx : filename.ml \
+@@ -253,6 +246,7 @@ stdlib__Filename.cmx : filename.ml \
      stdlib__Printf.cmx \
      stdlib__List.cmx \
+     stdlib__Lazy.cmx \
 +    stdlib__Fun.cmx \
-     stdlib__Domain.cmx \
      stdlib__Buffer.cmx \
      stdlib__Filename.cmi
-@@ -356,12 +350,14 @@ stdlib__Gc.cmo : gc.ml \
+ stdlib__Filename.cmi : filename.mli
+@@ -318,12 +312,14 @@ stdlib__Gc.cmo : gc.ml \
      stdlib__String.cmi \
      stdlib__Printf.cmi \
      stdlib__Printexc.cmi \
@@ -6502,23 +6503,23 @@
      stdlib__Gc.cmi
  stdlib__Gc.cmi : gc.mli \
      stdlib__Printexc.cmi
-@@ -373,6 +369,7 @@ stdlib__Hashtbl.cmo : hashtbl.ml \
+@@ -335,6 +331,7 @@ stdlib__Hashtbl.cmo : hashtbl.ml \
      stdlib__Obj.cmi \
+     stdlib__Lazy.cmi \
      stdlib__Int.cmi \
-     stdlib__Domain.cmi \
 +    stdlib__Atomic.cmi \
      stdlib__Array.cmi \
      stdlib__Hashtbl.cmi
  stdlib__Hashtbl.cmx : hashtbl.ml \
-@@ -383,6 +380,7 @@ stdlib__Hashtbl.cmx : hashtbl.ml \
+@@ -345,6 +342,7 @@ stdlib__Hashtbl.cmx : hashtbl.ml \
      stdlib__Obj.cmx \
+     stdlib__Lazy.cmx \
      stdlib__Int.cmx \
-     stdlib__Domain.cmx \
 +    stdlib__Atomic.cmx \
      stdlib__Array.cmx \
      stdlib__Hashtbl.cmi
  stdlib__Hashtbl.cmi : hashtbl.mli \
-@@ -717,8 +715,11 @@ stdlib__StdLabels.cmi : stdLabels.mli \
+@@ -663,8 +661,10 @@ stdlib__StdLabels.cmi : stdLabels.mli \
      stdlib__ListLabels.cmi \
      stdlib__BytesLabels.cmi \
      stdlib__ArrayLabels.cmi
@@ -6528,7 +6529,6 @@
 +    std_exit.cmi
 +std_exit.cmx : \
 +    std_exit.cmi
-+std_exit.cmi :
  stdlib__String.cmo : string.ml \
      stdlib.cmi \
      stdlib__Bytes.cmi \
