--- a	2023-09-28 18:46:12.269657895 +0100
+++ b	2023-09-28 18:46:12.289658654 +0100
@@ -12,7 +12,7 @@
  OCaml 5.0
  ---------
  
-@@ -772,6 +768,10 @@ OCaml 5.0
+@@ -767,6 +763,10 @@ OCaml 5.0
    the bug was only present in development version of OCaml 5.
    (Stephen Dolan, report by Andre Maroneze, review by Gabriel Scherer)
  
@@ -20,6 +20,6 @@
 +  multiple threads.
 +  (Marc Lasson, Nicolás Ojeda Bär, review by Gabriel Scherer and David Allsopp)
 +
- OCaml 4.14.0 (28 March 2022)
- ----------------------------
+ OCaml 4.14 maintenance branch
+ -----------------------------
  
