--- a	2023-09-28 18:43:15.994610604 +0100
+++ b	2023-09-28 18:43:16.038612465 +0100
@@ -275,10 +275,6 @@
  ### Build system:
  
  - #10717: Simplify the installation of man pages
-diff --git a/boot/ocamlc b/boot/ocamlc
-Binary files a/boot/ocamlc and b/boot/ocamlc differ
-diff --git a/boot/ocamllex b/boot/ocamllex
-Binary files a/boot/ocamllex and b/boot/ocamllex differ
 diff --git a/compilerlibs/Makefile.compilerlibs b/compilerlibs/Makefile.compilerlibs
 --- a/compilerlibs/Makefile.compilerlibs
 +++ b/compilerlibs/Makefile.compilerlibs
