--- a	2023-09-28 18:47:32.268627421 +0100
+++ b	2023-09-28 18:47:32.284628003 +0100
@@ -16,7 +16,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -1924,7 +1924,8 @@ AC_CHECK_FUNC([ffs], [AC_DEFINE([HAS_FFS])])
+@@ -1938,7 +1938,8 @@ AC_CHECK_FUNC([ffs], [AC_DEFINE([HAS_FFS])])
  AC_CHECK_FUNC([_BitScanForward], [AC_DEFINE([HAS_BITSCANFORWARD])])
  
  ## pthread_getaffinity_np, args differ from GNU and BSD
@@ -26,7 +26,7 @@
    [AC_LANG_PROGRAM(
      [[#define _GNU_SOURCE
        #include <sched.h>
-@@ -1933,8 +1934,9 @@ AC_COMPILE_IFELSE(
+@@ -1947,8 +1948,9 @@ AC_COMPILE_IFELSE(
        CPU_ZERO(&cs);
        CPU_COUNT(&cs);
        pthread_getaffinity_np(pthread_self(), sizeof(cs), &cs);]])],
@@ -38,7 +38,7 @@
      [AC_LANG_PROGRAM(
        [[#include <pthread.h>
          #include <pthread_np.h>
-@@ -1944,7 +1946,9 @@ AC_COMPILE_IFELSE(
+@@ -1958,7 +1960,9 @@ AC_COMPILE_IFELSE(
          CPU_ZERO(&cs);
          CPU_COUNT(&cs);
          pthread_getaffinity_np(pthread_self(), sizeof(cs), &cs);]])],
