--- a	2023-09-28 18:46:22.274036139 +0100
+++ b	2023-09-28 18:46:22.278036290 +0100
@@ -1,7 +1,7 @@
 diff --git a/asmcomp/amd64/emit.mlp b/asmcomp/amd64/emit.mlp
 --- a/asmcomp/amd64/emit.mlp
 +++ b/asmcomp/amd64/emit.mlp
-@@ -903,20 +903,21 @@ let fundecl fundecl =
+@@ -953,22 +953,23 @@ let fundecl fundecl =
      preproc_stack_check
        ~fun_body:fundecl.fun_body ~frame_size:(frame_size env) ~trap_size:16
    in
@@ -26,9 +26,11 @@
 +      Some (overflow, ret)
 +    end else None
 +  in
+ *)
    emit_all env true fundecl.fun_body;
    List.iter emit_call_gc env.call_gc_sites;
    emit_call_bound_errors env;
+ (* BACKPORT
 -  begin match !handle_overflow with
 +  begin match handle_overflow with
    | None -> ()
@@ -37,7 +39,7 @@
 diff --git a/asmcomp/arm64/emit.mlp b/asmcomp/arm64/emit.mlp
 --- a/asmcomp/arm64/emit.mlp
 +++ b/asmcomp/arm64/emit.mlp
-@@ -1076,24 +1076,23 @@ let fundecl fundecl =
+@@ -1127,26 +1127,25 @@ let fundecl fundecl =
      preproc_stack_check
        ~fun_body:fundecl.fun_body ~frame_size:(frame_size env) ~trap_size:16
    in
@@ -70,17 +72,19 @@
 +      Some (overflow, ret), 5
 +    end else None, 0
 +  in
+ *)
  
    let max_out_of_line_code_offset =
+ (* BACKPORT
 -    !stack_check_size +
 +    stack_check_size +
+ *)
      max_out_of_line_code_offset ~num_call_gc
        ~num_check_bound
-   in
-@@ -1106,7 +1105,7 @@ let fundecl fundecl =
-   assert (List.length env.call_gc_sites = num_call_gc);
+@@ -1161,7 +1160,7 @@ let fundecl fundecl =
    assert (List.length env.bound_error_sites = num_check_bound);
  
+ (* BACKPORT
 -  begin match !handle_overflow with
 +  begin match handle_overflow with
    | None -> ()
@@ -89,13 +93,14 @@
 diff --git a/asmcomp/riscv/emit.mlp b/asmcomp/riscv/emit.mlp
 --- a/asmcomp/riscv/emit.mlp
 +++ b/asmcomp/riscv/emit.mlp
-@@ -617,24 +617,25 @@ let fundecl fundecl =
+@@ -624,19 +624,19 @@ let fundecl fundecl =
      preproc_stack_check
        ~fun_body:fundecl.fun_body ~frame_size:(frame_size env) ~trap_size:16
    in
 -  let handle_overflow = ref None in
 -  if contains_nontail_calls || max_frame_size >= stack_threshold_size then begin
 -    let overflow = new_label () and ret = new_label () in
+-
 -    let threshold_offset = Domainstate.stack_ctx_words * 8 + stack_threshold_size in
 -    let f = max_frame_size + threshold_offset in
 -    let offset = Domainstate.(idx_of_field Domain_current_stack) * 8 in
@@ -118,11 +123,13 @@
 +      Some (overflow, ret)
 +    end else None
 +  in
+ *)
  
    emit_all env fundecl.fun_body;
-   List.iter emit_call_gc env.call_gc_sites;
+@@ -644,7 +644,7 @@ let fundecl fundecl =
    List.iter emit_call_bound_error env.bound_error_sites;
  
+ (* BACKPORT
 -  begin match !handle_overflow with
 +  begin match handle_overflow with
    | None -> ()
