--- a	2023-09-28 18:45:49.584792436 +0100
+++ b	2023-09-28 18:45:49.588792590 +0100
@@ -1,7 +1,7 @@
 diff --git a/stdlib/format.mli b/stdlib/format.mli
 --- a/stdlib/format.mli
 +++ b/stdlib/format.mli
-@@ -1285,7 +1285,7 @@ val fprintf : formatter -> ('a, formatter, unit) format -> 'a
+@@ -1288,7 +1288,7 @@ val fprintf : formatter -> ('a, formatter, unit) format -> 'a
      optionally specified with the following syntax:
      the [<] character, followed by an integer [nspaces] value,
      then an integer [offset], and a closing [>] character.
