--- a	2023-09-28 18:45:48.252741274 +0100
+++ b	2023-09-28 18:45:48.260741581 +0100
@@ -28,31 +28,29 @@
 -### 4.12 and 4.13. Indeed, for OCaml 4.12 most of the time the variable
 -### contained -lpthread. From 4.13 onward it will most of the time be
 -### empty since we have -pthread in CFLAGS which implies -lpthread.
-diff --git a/configure b/configure
-Binary files a/configure and b/configure differ
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -1984,7 +1984,7 @@ AS_CASE([$enable_debug_runtime],
+@@ -2007,7 +2007,7 @@ AS_CASE([$arch,$system],
  
  AS_CASE([$host],
    [*-*-mingw32|*-pc-windows],
--    [PTHREAD_LIBS="-lpthread"],
-+    [PTHREAD_LIBS="-l:libpthread.a"],
+-dnl    [PTHREAD_LIBS="-lpthread"],
++dnl    [PTHREAD_LIBS="-l:libpthread.a"],
+     [PTHREAD_LIBS=''],
    [AX_PTHREAD(
      [common_cflags="$common_cflags $PTHREAD_CFLAGS"
-     saved_CFLAGS="$CFLAGS"
-@@ -2182,8 +2182,8 @@ AC_CHECK_LIB(execinfo, backtrace, cclibs="$cclibs -lexecinfo",[])
+@@ -2230,8 +2230,8 @@ AC_CHECK_LIB(execinfo, backtrace, cclibs="$cclibs -lexecinfo",[])
  
  AS_CASE([$host],
    [*-*-mingw32],
--    [bytecclibs="-lws2_32 -lversion -lpthread -lgcc_eh -lDbgHelp"
--    nativecclibs="-lws2_32 -lversion -lpthread -lgcc_eh -lDbgHelp"],
-+    [bytecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh -lDbgHelp"
-+    nativecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh -lDbgHelp"],
+-dnl    [bytecclibs="-lws2_32 -lversion -lpthread -lgcc_eh -lDbgHelp"
+-dnl    nativecclibs="-lws2_32 -lversion -lpthread -lgcc_eh -lDbgHelp"],
++dnl    [bytecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh -lDbgHelp"
++dnl    nativecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh -lDbgHelp"],
+     [bytecclibs="-lws2_32 -lversion -lgcc_eh -lDbgHelp"
+     nativecclibs="-lws2_32 -lversion -lgcc_eh -lDbgHelp"],
    [*-pc-windows],
-     [bytecclibs="advapi32.lib ws2_32.lib version.lib"
-     nativecclibs="advapi32.lib ws2_32.lib version.lib"],
 diff --git a/otherlibs/str/Makefile b/otherlibs/str/Makefile
 --- a/otherlibs/str/Makefile
 +++ b/otherlibs/str/Makefile
