--- a	2023-09-28 18:46:54.707248505 +0100
+++ b	2023-09-28 18:46:54.723249098 +0100
@@ -88,7 +88,7 @@
 diff --git a/testsuite/tests/lib-runtime-events/test_caml_counters.ml b/testsuite/tests/lib-runtime-events/test_caml_counters.ml
 --- a/testsuite/tests/lib-runtime-events/test_caml_counters.ml
 +++ b/testsuite/tests/lib-runtime-events/test_caml_counters.ml
-@@ -12,9 +12,19 @@ let () =
+@@ -14,9 +14,19 @@ let () =
      start ();
      let cursor = create_cursor None in
      let callbacks = Callbacks.create ~runtime_counter () in
