--- a	2023-09-28 18:46:51.263120743 +0100
+++ b	2023-09-28 18:46:51.279121337 +0100
@@ -16,7 +16,7 @@
 diff --git a/stdlib/.depend b/stdlib/.depend
 --- a/stdlib/.depend
 +++ b/stdlib/.depend
-@@ -611,6 +611,7 @@ stdlib__Queue.cmx : queue.ml \
+@@ -562,6 +562,7 @@ stdlib__Queue.cmx : queue.ml \
  stdlib__Queue.cmi : queue.mli \
      stdlib__Seq.cmi
  stdlib__Random.cmo : random.ml \
@@ -24,7 +24,7 @@
      stdlib__String.cmi \
      stdlib__Nativeint.cmi \
      stdlib__Int64.cmi \
-@@ -622,6 +623,7 @@ stdlib__Random.cmo : random.ml \
+@@ -572,6 +573,7 @@ stdlib__Random.cmo : random.ml \
      stdlib__Array.cmi \
      stdlib__Random.cmi
  stdlib__Random.cmx : random.ml \
