--- a	2023-09-28 18:48:28.878658672 +0100
+++ b	2023-09-28 18:48:28.894659239 +0100
@@ -25,7 +25,7 @@
 diff --git a/testsuite/tests/lib-bigarray/bigarrays.ml b/testsuite/tests/lib-bigarray/bigarrays.ml
 --- a/testsuite/tests/lib-bigarray/bigarrays.ml
 +++ b/testsuite/tests/lib-bigarray/bigarrays.ml
-@@ -1164,7 +1164,7 @@ let tests () =
+@@ -1166,7 +1166,7 @@ let tests () =
    test_structured_io 13 (make_array2 complex32 c_layout 0 100 100 makecomplex);
    test_structured_io 14 (make_array3 complex64 fortran_layout 1 10 20 30
                                       makecomplex);
