--- a	2023-09-28 18:48:08.269925427 +0100
+++ b	2023-09-28 18:48:08.285925999 +0100
@@ -1,7 +1,7 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -640,6 +640,10 @@ Some of those changes will benefit all OCaml packages.
+@@ -641,6 +641,10 @@ Some of those changes will benefit all OCaml packages.
    `pp_print_newline` flushes its newline
    (Florian Angeletti, review by Gabriel Scherer)
  
