--- a	2023-09-28 18:47:05.707655057 +0100
+++ b	2023-09-28 18:47:05.731655941 +0100
@@ -68,10 +68,6 @@
  ### Build system:
  
  - #11590: Allow installing to a destination path containing spaces.
-diff --git a/boot/ocamlc b/boot/ocamlc
-Binary files a/boot/ocamlc and b/boot/ocamlc differ
-diff --git a/boot/ocamllex b/boot/ocamllex
-Binary files a/boot/ocamllex and b/boot/ocamllex differ
 diff --git a/bytecomp/emitcode.ml b/bytecomp/emitcode.ml
 --- a/bytecomp/emitcode.ml
 +++ b/bytecomp/emitcode.ml
@@ -257,7 +253,7 @@
          | _ -> print_string "<wrong reloc>"
      with Not_found ->
        print_string "<no reloc>"
-@@ -484,7 +450,7 @@ let print_code ic len =
+@@ -486,7 +452,7 @@ let print_code ic len =
  let print_reloc (info, pos) =
    printf "    %d    (%d)    " pos (pos/4);
    match info with
