--- a	2023-09-28 18:44:29.993663815 +0100
+++ b	2023-09-28 18:44:30.021664943 +0100
@@ -169,10 +169,6 @@
  # Variables used to represent the OCaml runtime system
  # Most of the time, boot/ocamlrun and runtime/ocamlrun are the same.
  # However, under some circumstances it is important to be able to
-diff --git a/boot/ocamlc b/boot/ocamlc
-Binary files a/boot/ocamlc and b/boot/ocamlc differ
-diff --git a/boot/ocamllex b/boot/ocamllex
-Binary files a/boot/ocamllex and b/boot/ocamllex differ
 diff --git a/compilerlibs/Makefile.compilerlibs b/compilerlibs/Makefile.compilerlibs
 --- a/compilerlibs/Makefile.compilerlibs
 +++ b/compilerlibs/Makefile.compilerlibs
@@ -239,7 +235,7 @@
 new file mode 100644
 --- /dev/null
 +++ b/utils/config.common.ml
-@@ -0,0 +1,162 @@
+@@ -0,0 +1,167 @@
 +#2 "utils/config.common.ml"
 +(**************************************************************************)
 +(*                                                                        *)
@@ -304,8 +300,13 @@
 +let lazy_tag = 246
 +
 +let max_young_wosize = 256
++(* BACKPORT BEGIN
 +let stack_threshold = 32 (* see runtime/caml/config.h *)
 +let stack_safety_margin = 6
++*)
++let stack_threshold = 256 (* see runtime/caml/config.h *)
++let stack_safety_margin = 60
++(* BACKPORT END *)
 +let default_executable_name =
 +  match Sys.os_type with
 +    "Unix" -> "a.out"
@@ -514,7 +515,7 @@
  let force_instrumented_runtime = %%FORCE_INSTRUMENTED_RUNTIME%%
  
  let flat_float_array = %%FLAT_FLOAT_ARRAY%%
-@@ -91,39 +76,6 @@ let flat_float_array = %%FLAT_FLOAT_ARRAY%%
+@@ -91,44 +76,6 @@ let flat_float_array = %%FLAT_FLOAT_ARRAY%%
  let function_sections = %%FUNCTION_SECTIONS%%
  let afl_instrument = %%AFL_INSTRUMENT%%
  
@@ -548,13 +549,18 @@
 -let lazy_tag = 246
 -
 -let max_young_wosize = 256
+-(* BACKPORT BEGIN
 -let stack_threshold = 32 (* see runtime/caml/config.h *)
 -let stack_safety_margin = 6
+-*)
+-let stack_threshold = 256 (* see runtime/caml/config.h *)
+-let stack_safety_margin = 60
+-(* BACKPORT END *)
 -
  let architecture = "%%ARCH%%"
  let model = "%%MODEL%%"
  let system = "%%SYSTEM%%"
-@@ -143,104 +95,6 @@ let ext_dll = "%%EXT_DLL%%"
+@@ -148,104 +95,6 @@ let ext_dll = "%%EXT_DLL%%"
  let host = "%%HOST%%"
  let target = "%%TARGET%%"
  
