--- a	2023-09-28 18:44:20.545282171 +0100
+++ b	2023-09-28 18:44:20.581283630 +0100
@@ -178,7 +178,7 @@
  AC_SUBST([outputexe])
  AC_SUBST([outputobj])
  AC_SUBST([syslib])
-@@ -533,11 +532,8 @@ AS_CASE([$ocaml_cv_cc_vendor],
+@@ -543,11 +542,8 @@ AS_CASE([$ocaml_cv_cc_vendor],
  
  AS_CASE([$host],
    [*-*-mingw32|*-pc-windows],
@@ -192,7 +192,7 @@
  AS_CASE([$host],
    [*-*-cygwin*|*-*-mingw32|*-pc-windows],
      [exeext=".exe"],
-@@ -546,7 +542,7 @@ AS_CASE([$host],
+@@ -556,7 +552,7 @@ AS_CASE([$host],
  otherlibraries="dynlink"
  AS_IF([test x"$enable_unix_lib" != "xno"],
    [enable_unix_lib=yes
@@ -1030,7 +1030,7 @@
  #define Nothing ((value) 0)
  
  extern value unix_error_of_code (int errcode);
-@@ -43,18 +96,30 @@ extern void caml_unix_check_path(value path, const char * cmdname);
+@@ -43,19 +96,31 @@ extern void caml_unix_check_path(value path, const char * cmdname);
  
  #define DIR_Val(v) *((DIR **) &Field(v, 0))
  
@@ -1062,7 +1062,8 @@
  #define EXECV_CAST
 +#endif
  
- #endif /* CAML_UNIXSUPPORT_H */
+ /* BACKPORT BEGIN */
+ #define caml_process_pending_signals caml_process_pending_actions
 diff --git a/otherlibs/unix/unixsupport.c b/otherlibs/unix/unixsupport_unix.c
 similarity index 100%
 rename from otherlibs/unix/unixsupport.c
