--- a	2023-09-28 18:44:13.749006227 +0100
+++ b	2023-09-28 18:44:13.749006227 +0100
@@ -1,16 +0,0 @@
-diff --git a/configure b/configure
-Binary files a/configure and b/configure differ
-diff --git a/configure.ac b/configure.ac
---- a/configure.ac
-+++ b/configure.ac
-@@ -1111,8 +1111,8 @@ dnl [x86_64-*-gnu*],
- dnl   [arch=amd64; system=gnu],
- dnl [x86_64-*-dragonfly*],
- dnl   [arch=amd64; system=dragonfly],
--dnl [x86_64-*-solaris*],
--dnl   [arch=amd64; system=solaris],
-+  [x86_64-*-solaris*],
-+    [arch=amd64; system=solaris],
-   [x86_64-*-freebsd*],
-     [arch=amd64; system=freebsd],
-   [x86_64-*-netbsd*],
