--- a	2023-09-28 18:47:53.337389745 +0100
+++ b	2023-09-28 18:47:53.345390033 +0100
@@ -23,7 +23,7 @@
 diff --git a/stdlib/StdlibModules b/stdlib/StdlibModules
 --- a/stdlib/StdlibModules
 +++ b/stdlib/StdlibModules
-@@ -104,9 +104,23 @@ STDLIB_MODULE_BASENAMES = \
+@@ -99,9 +99,23 @@ STDLIB_MODULE_BASENAMES = \
  STDLIB_PREFIXED_MODULES = \
    $(filter-out stdlib camlinternal%, $(STDLIB_MODULE_BASENAMES))
  
