--- a	2023-09-28 18:48:14.466146613 +0100
+++ b	2023-09-28 18:48:14.478147040 +0100
@@ -1,7 +1,7 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -959,6 +959,11 @@ Some of those changes will benefit all OCaml packages.
+@@ -960,6 +960,11 @@ Some of those changes will benefit all OCaml packages.
    with ZSTD support.
    (David Allsopp, review by Sébastien Hinderer)
  
@@ -18,7 +18,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -1146,6 +1146,9 @@ AS_IF([test x"$enable_shared" != "xno"],
+@@ -1159,6 +1159,9 @@ AS_IF([test x"$enable_shared" != "xno"],
             # See https://github.com/ocaml/ocaml/issues/9800
             [mkdll_flags='-shared -Wl,-z,notext'],
             [mkdll_flags='-shared'])
@@ -28,7 +28,7 @@
        oc_ldflags="$oc_ldflags -Wl,-E"
        rpath="-Wl,-rpath,"
        mksharedlibrpath="-Wl,-rpath,"
-@@ -1160,6 +1163,14 @@ AS_IF([test -z "$mkmaindll"],
+@@ -1173,6 +1176,14 @@ AS_IF([test -z "$mkmaindll"],
    [mkmaindll_exp="$mkdll_exp"
    mkmaindll="$mkdll"])
  
