--- a	2023-09-28 18:47:59.025594246 +0100
+++ b	2023-09-28 18:47:59.041594820 +0100
@@ -15,8 +15,8 @@
 diff --git a/asmcomp/amd64/emit.mlp b/asmcomp/amd64/emit.mlp
 --- a/asmcomp/amd64/emit.mlp
 +++ b/asmcomp/amd64/emit.mlp
-@@ -79,14 +79,12 @@ let fp = Config.with_frame_pointers
- let stack_threshold_size = Config.stack_threshold * 8 (* bytes *)
+@@ -83,14 +83,12 @@ let stack_threshold_size = Config.stack_threshold * 8 (* bytes *)
+ *)
  
  let frame_size env =                     (* includes return address *)
 -  if env.f.fun_frame_required then begin
