--- a	2023-09-28 18:44:02.144532243 +0100
+++ b	2023-09-28 18:44:02.160532899 +0100
@@ -61,7 +61,7 @@
 diff --git a/otherlibs/unix/signals.c b/otherlibs/unix/signals.c
 --- a/otherlibs/unix/signals.c
 +++ b/otherlibs/unix/signals.c
-@@ -30,10 +30,9 @@
+@@ -36,10 +36,9 @@
  static void decode_sigset(value vset, sigset_t * set)
  {
    sigemptyset(set);
@@ -73,15 +73,17 @@
    }
  }
  
-@@ -45,7 +44,7 @@ static value encode_sigset(sigset_t * set)
- 
-   for (i = 1; i < NSIG; i++)
-     if (sigismember(set, i) > 0) {
+@@ -55,9 +54,8 @@ static value encode_sigset(sigset_t * set)
+       value newcons = caml_alloc_small(2, 0);
+       Field(newcons, 0) = Val_int(caml_rev_convert_signal_number(i));
+       Field(newcons, 1) = res;
+-/* BACKPORT END */
+ #if 0 /* BACKPORT */
 -      value newcons = caml_alloc_2(0,
 +      value newcons = caml_alloc_2(Tag_cons,
          Val_int(caml_rev_convert_signal_number(i)),
          res);
-       res = newcons;
+ #endif
 diff --git a/otherlibs/win32unix/createprocess.c b/otherlibs/win32unix/createprocess.c
 --- a/otherlibs/win32unix/createprocess.c
 +++ b/otherlibs/win32unix/createprocess.c
