--- a	2023-09-28 18:47:06.843696913 +0100
+++ b	2023-09-28 18:47:06.863697649 +0100
@@ -44,7 +44,7 @@
  
  
  (* To get a newline when one does not want to close the current box. *)
-@@ -1077,7 +1077,7 @@ let get_stdbuf () = DLS.get stdbuf_key
+@@ -1089,7 +1089,7 @@ let get_stdbuf () = DLS.get stdbuf_key
     Formatter [ppf] is supposed to print to buffer [buf], otherwise this
     function is not really useful. *)
  let flush_buffer_formatter buf ppf =
