--- a	2023-09-28 18:47:54.585434661 +0100
+++ b	2023-09-28 18:47:54.609435524 +0100
@@ -32,7 +32,7 @@
  (* Complex *)
  
  let box_complex dbg c_re c_im =
-@@ -852,6 +859,7 @@ let curry_function_sym n =
+@@ -856,6 +863,7 @@ let curry_function_sym n =
  
  let bigarray_elt_size : Lambda.bigarray_kind -> int = function
      Pbigarray_unknown -> assert false
@@ -40,7 +40,7 @@
    | Pbigarray_float32 -> 4
    | Pbigarray_float64 -> 8
    | Pbigarray_sint8 -> 1
-@@ -925,6 +933,7 @@ let bigarray_indexing unsafe elt_kind layout b args dbg =
+@@ -929,6 +937,7 @@ let bigarray_indexing unsafe elt_kind layout b args dbg =
  
  let bigarray_word_kind : Lambda.bigarray_kind -> memory_chunk = function
      Pbigarray_unknown -> assert false
