--- a	2023-09-28 18:46:47.594984418 +0100
+++ b	2023-09-28 18:46:47.614985162 +0100
@@ -9,6 +9,6 @@
 +  (Andrei Paskevich and Charlène Gros, review by David Allsopp and
 +   Nicolás Ojeda Bär)
 +
- OCaml 5.0
- ---------
- 
+ - #12264, #12289: Fix compact_allocate to avoid a pathological case
+   that causes very slow compaction.
+   (Damien Doligez, report by Arseniy Alekseyev, review by Sadiq Jaffer)
