--- a	2023-09-28 18:45:33.812184112 +0100
+++ b	2023-09-28 18:45:33.832184887 +0100
@@ -156,7 +156,7 @@
  
     If multiple domains write to the same output channel using the
     predefined formatters (as obtained by {!get_std_formatter} or
-@@ -1405,3 +1407,142 @@ val kasprintf : (string -> 'a) -> ('b, formatter, unit, 'a) format4 -> 'b
+@@ -1408,3 +1410,142 @@ val kasprintf : (string -> 'a) -> ('b, formatter, unit, 'a) format4 -> 'b
  
    @since 4.03
  *)
