--- a	2023-09-28 18:46:37.142594521 +0100
+++ b	2023-09-28 18:46:37.146594670 +0100
@@ -1,13 +1,15 @@
 diff --git a/testsuite/tests/memory-model/publish.ml b/testsuite/tests/memory-model/publish.ml
 --- a/testsuite/tests/memory-model/publish.ml
 +++ b/testsuite/tests/memory-model/publish.ml
-@@ -1,7 +1,8 @@
- (* TEST
-   modules="opt.ml barrier.ml hist.ml shared.ml run.ml outcome.ml"
-   * not-bsd
--  ** bytecode
-+  ** not-windows
-+  *** bytecode
-   ** native
+@@ -3,8 +3,9 @@
+   * skip
+   reason = "OCaml 5 only"
+   ** not-bsd
+-  *** bytecode
+-  *** native
++  *** not-windows
++  **** bytecode
++  **** native
  *)
  
+ (* Memory model: test the _publish idiom *)
