--- a	2023-09-28 18:43:28.883153339 +0100
+++ b	2023-09-28 18:43:28.891153674 +0100
@@ -3,7 +3,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -1062,99 +1062,107 @@ model=default
+@@ -1071,99 +1071,107 @@ model=default
  system=unknown
  
  AS_CASE([$host],
@@ -190,7 +190,7 @@
  
  AS_IF([! $native_compiler], [natdynlink=false])
  
-@@ -1799,37 +1807,35 @@ AS_CASE([$arch,$system],
+@@ -1808,38 +1816,35 @@ AS_CASE([$arch,$system],
      AC_MSG_RESULT([yes])],
    [AC_MSG_RESULT([no])])
  
@@ -226,7 +226,8 @@
 -    [*-*-mingw32|*-pc-windows],
 -      [systhread_support=true
 -      otherlibraries="$otherlibraries systhreads"
--      PTHREAD_LIBS="-lpthread"
+-dnl      PTHREAD_LIBS="-lpthread"
+-      PTHREAD_LIBS=''
 -      AC_MSG_NOTICE([the Win32 threads library is supported])],
 -    [AX_PTHREAD(
 -      [systhread_support=true
