--- a	2023-09-28 18:46:06.809450580 +0100
+++ b	2023-09-28 18:46:06.837451645 +0100
@@ -11,10 +11,6 @@
  ### Build system:
  - #11590: Allow installing to a destination path containing spaces.
     (Élie Brami, review by Sébastien Hinderer and David Allsopp)
-diff --git a/boot/ocamlc b/boot/ocamlc
-Binary files a/boot/ocamlc and b/boot/ocamlc differ
-diff --git a/boot/ocamllex b/boot/ocamllex
-Binary files a/boot/ocamllex and b/boot/ocamllex differ
 diff --git a/testsuite/tests/shadow_include/shadow_all.ml b/testsuite/tests/shadow_include/shadow_all.ml
 --- a/testsuite/tests/shadow_include/shadow_all.ml
 +++ b/testsuite/tests/shadow_include/shadow_all.ml
