--- a	2023-09-28 18:44:00.620469724 +0100
+++ b	2023-09-28 18:44:00.644470709 +0100
@@ -63,7 +63,7 @@
  let print_locations = ref true
  let print_reloc_info = ref false
  
-@@ -422,7 +423,7 @@ let print_instr ic =
+@@ -424,7 +425,7 @@ let print_instr ic =
    else print_string names_of_instructions.(op);
    begin try
      let shape = List.assoc op op_shapes in
@@ -72,7 +72,7 @@
      match shape with
      | Uint -> print_int (inputu ic)
      | Sint -> print_int (inputs ic)
-@@ -547,6 +548,7 @@ let dump_exe ic =
+@@ -549,6 +550,7 @@ let dump_exe ic =
    print_code ic code_size
  
  let arg_list = [
@@ -80,7 +80,7 @@
    "-noloc", Arg.Clear print_locations, " : don't print source information";
    "-reloc", Arg.Set print_reloc_info, " : print relocation information";
    "-args", Arg.Expand Arg.read_arg,
-@@ -566,14 +568,14 @@ let arg_fun filename =
+@@ -568,14 +570,14 @@ let arg_fun filename =
    let ic = open_in_bin filename in
    if not !first_file then print_newline ();
    first_file := false;
