--- a	2023-09-28 18:48:06.037845594 +0100
+++ b	2023-09-28 18:48:06.053846166 +0100
@@ -1,16 +1,37 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -8,6 +8,10 @@ OCaml 4.14 maintenance version
+@@ -1600,6 +1600,11 @@ OCaml 5.0.0 (15 December 2022)
+   (Damien Doligez, report by Thierry Martinez and Vincent Laviron, review by
+    Xavier Leroy)
  
- ### Bug fixes:
++- #11508, #11509: make Bytes.escaped domain-safe
++  (Christiano Haesbaert and Gabriel Scherer,
++   review by Xavier Leroy,
++   report by Jan Midtgaard and Tom Kelly)
++
+ - #11516, #11524: Fix the `deprecated_mutable` attribute.
+   (Chris Casinghino, review by Nicolás Ojeda Bär and Florian Angeletti)
+ 
+@@ -1631,10 +1636,15 @@ OCaml 5.0.0 (15 December 2022)
+ OCaml 4.14 maintenance branch
+ -----------------------------
  
+-- #11508, #11509: make Bytes.escaped domain-safe
+-  (Christiano Haesbaert and Gabriel Scherer,
+-   review by Xavier Leroy,
+-   report by Jan Midtgaard and Tom Kelly)
++### Bug fixes:
++
 +- #12264, #12289: Fix compact_allocate to avoid a pathological case
 +  that causes very slow compaction.
 +  (Damien Doligez, report by Arseniy Alekseyev, review by Sadiq Jaffer)
 +
- OCaml 4.14.1 (20 December 2022)
- ------------------------------
++
++OCaml 4.14.1 (20 December 2022)
++------------------------------
+ 
+ ### Manual and documentation:
  
 diff --git a/runtime/compact.c b/runtime/compact.c
 --- a/runtime/compact.c
