--- a	2023-09-28 18:44:33.029785966 +0100
+++ b	2023-09-28 18:44:33.037786287 +0100
@@ -1,7 +1,7 @@
 diff --git a/Makefile b/Makefile
 --- a/Makefile
 +++ b/Makefile
-@@ -934,8 +934,8 @@ $(foreach runtime_OBJECT_TYPE, $(runtime_OBJECT_TYPES), \
+@@ -931,8 +931,8 @@ $(foreach runtime_OBJECT_TYPE, $(runtime_OBJECT_TYPES), \
  runtime/$(UNIX_OR_WIN32)_non_shared.%.$(O): \
    OC_CPPFLAGS += -DBUILDING_LIBCAMLRUNS
  
