--- a	2023-09-28 18:44:24.997462293 +0100
+++ b	2023-09-28 18:44:25.001462455 +0100
@@ -1,7 +1,7 @@
 diff --git a/asmcomp/arm64/emit.mlp b/asmcomp/arm64/emit.mlp
 --- a/asmcomp/arm64/emit.mlp
 +++ b/asmcomp/arm64/emit.mlp
-@@ -1085,7 +1085,7 @@ let fundecl fundecl =
+@@ -1136,7 +1136,7 @@ let fundecl fundecl =
      let f = max_frame_size + threshold_offset in
      let offset = Domainstate.(idx_of_field Domain_current_stack) * 8 in
      `  ldr {emit_reg reg_tmp1}, [{emit_reg reg_domain_state_ptr}, #{emit_int offset}]\n`;
