--- a	2023-09-28 18:47:46.073127779 +0100
+++ b	2023-09-28 18:47:46.077127924 +0100
@@ -13,7 +13,7 @@
 diff --git a/testsuite/tests/lib-sync/trylock.ml b/testsuite/tests/lib-sync/trylock.ml
 --- a/testsuite/tests/lib-sync/trylock.ml
 +++ b/testsuite/tests/lib-sync/trylock.ml
-@@ -6,6 +6,7 @@ let () =
+@@ -9,6 +9,7 @@ let () =
    let m = Mutex.create () in
    Mutex.lock m;
    let res = Mutex.try_lock m in
