--- a	2023-09-28 18:45:56.133043392 +0100
+++ b	2023-09-28 18:45:56.141043698 +0100
@@ -3,14 +3,18 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -2166,8 +2166,8 @@ AC_CHECK_LIB(execinfo, backtrace, cclibs="$cclibs -lexecinfo",[])
+@@ -2214,10 +2214,10 @@ AC_CHECK_LIB(execinfo, backtrace, cclibs="$cclibs -lexecinfo",[])
  
  AS_CASE([$host],
    [*-*-mingw32],
--    [bytecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh -lDbgHelp"
--    nativecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh -lDbgHelp"],
-+    [bytecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh"
-+    nativecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh"],
+-dnl    [bytecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh -lDbgHelp"
+-dnl    nativecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh -lDbgHelp"],
+-    [bytecclibs="-lws2_32 -lversion -lgcc_eh -lDbgHelp"
+-    nativecclibs="-lws2_32 -lversion -lgcc_eh -lDbgHelp"],
++dnl    [bytecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh"
++dnl    nativecclibs="-lws2_32 -lversion -l:libpthread.a -lgcc_eh"],
++    [bytecclibs="-lws2_32 -lversion -lgcc_eh"
++    nativecclibs="-lws2_32 -lversion -lgcc_eh"],
    [*-pc-windows],
      [bytecclibs="advapi32.lib ws2_32.lib version.lib"
      nativecclibs="advapi32.lib ws2_32.lib version.lib"],
