--- a	2023-09-28 18:44:42.854179634 +0100
+++ b	2023-09-28 18:44:42.862179954 +0100
@@ -1,7 +1,7 @@
 diff --git a/testsuite/tests/lib-runtime-events/test_caml.ml b/testsuite/tests/lib-runtime-events/test_caml.ml
 --- a/testsuite/tests/lib-runtime-events/test_caml.ml
 +++ b/testsuite/tests/lib-runtime-events/test_caml.ml
-@@ -25,7 +25,7 @@ let lifecycle domain_id ts lifecycle_event data =
+@@ -27,7 +27,7 @@ let lifecycle domain_id ts lifecycle_event data =
  
  let runtime_begin domain_id ts phase =
      match phase with
@@ -10,7 +10,7 @@
          begin
              assert(!major == 0);
              major := 1
-@@ -39,7 +39,7 @@ let runtime_begin domain_id ts phase =
+@@ -41,7 +41,7 @@ let runtime_begin domain_id ts phase =
  
  let runtime_end domain_id ts phase =
      match phase with
@@ -19,7 +19,7 @@
          begin
              assert(!major == 1);
              major := 0;
-@@ -76,4 +76,4 @@ let () =
+@@ -78,4 +78,4 @@ let () =
          ignore(read_poll cursor callbacks None)
      done;
      assert(!got_start);
