--- a	2023-09-28 18:46:24.314113018 +0100
+++ b	2023-09-28 18:46:24.330113620 +0100
@@ -27,7 +27,7 @@
  
  val pp_set_geometry : formatter -> max_indent:int -> margin:int -> unit
  val set_geometry : max_indent:int -> margin:int -> unit
-@@ -983,6 +985,7 @@ val std_formatter : formatter
+@@ -984,6 +986,7 @@ val std_formatter : formatter
  val get_std_formatter : unit -> formatter
  (** [get_std_formatter ()] returns the current domain's standard formatter used
      to write to standard output.
@@ -35,7 +35,7 @@
  *)
  
  val err_formatter : formatter
-@@ -992,8 +995,9 @@ val err_formatter : formatter
+@@ -993,8 +996,9 @@ val err_formatter : formatter
  *)
  
  val get_err_formatter : unit -> formatter
@@ -46,7 +46,7 @@
  *)
  
  val formatter_of_buffer : Buffer.t -> formatter
-@@ -1008,7 +1012,8 @@ val stdbuf : Buffer.t
+@@ -1009,7 +1013,8 @@ val stdbuf : Buffer.t
  
  val get_stdbuf : unit -> Buffer.t
  (** [get_stdbuf ()] returns the current domain's string buffer in which the
@@ -56,7 +56,7 @@
  
  val str_formatter : formatter
  (** The initial domain's formatter to output to the {!stdbuf} string buffer.
-@@ -1019,6 +1024,7 @@ val str_formatter : formatter
+@@ -1020,6 +1025,7 @@ val str_formatter : formatter
  val get_str_formatter : unit -> formatter
  (** The current domain's formatter to output to the current domains string
      buffer.
@@ -64,14 +64,14 @@
  *)
  
  val flush_str_formatter : unit -> string
-@@ -1050,6 +1056,7 @@ val make_synchronized_formatter :
+@@ -1052,6 +1058,7 @@ val make_synchronized_formatter :
      When the formatter is used with multiple domains, the output from the
      domains will be interleaved with each other at points where the formatter
      is flushed, such as with {!print_flush}.
 +    @since 5.0
  *)
+ *)
  
- val formatter_of_out_functions :
 diff --git a/stdlib/sys.mli b/stdlib/sys.mli
 --- a/stdlib/sys.mli
 +++ b/stdlib/sys.mli
