--- a	2023-09-28 18:44:14.561039260 +0100
+++ b	2023-09-28 18:44:14.589040399 +0100
@@ -287,7 +287,7 @@
  .SECONDARY: $(lexers:.mll=.ml) $(parsers:.mly=.mli) $(parsers:.mly=.ml)
  
  .PHONY: all allopt opt.opt # allopt and opt.opt are synonyms
-@@ -312,7 +308,6 @@ $(DEPDIR)/%.$(D): %.c | $(DEPDIR)
+@@ -313,7 +309,6 @@ $(DEPDIR)/%.$(D): %.c | $(DEPDIR)
  
  .PHONY: depend
  depend: $(dependencies_generated_prereqs)
@@ -332,7 +332,7 @@
  
  OC_CPPFLAGS += -I$(ROOTDIR)/runtime
  
-@@ -262,14 +264,18 @@ GNUISH_SED = \
+@@ -262,15 +264,18 @@ GNUISH_SED = \
    $(if $(filter X,$(shell echo x | $(SED) -E -e 's/./\u&/' 2>/dev/null)),\
         $(SED),$(error GNU sed is needed for make depend))
  
@@ -345,7 +345,8 @@
 +
  .PHONY: depend
 -depend:
--	$(CAMLDEP) $(DEPFLAGS) $(filter-out stdlib.%,$(wildcard *.mli *.ml)) \
+-	$(CAMLDEP) $(DEPFLAGS) $(filter-out stdlib.%,std_exit.ml \
+-    $(STDLIB_MODULE_BASENAMES:=.mli) $(STDLIB_MODULE_BASENAMES:=.ml)) \
 -	  > .depend.tmp
 -	$(CAMLDEP) $(DEPFLAGS) -pp "$(AWK) -f ./remove_module_aliases.awk" \
 -	  stdlib.ml stdlib.mli >> .depend.tmp
