--- a	2023-09-28 18:48:29.438678500 +0100
+++ b	2023-09-28 18:48:29.458679208 +0100
@@ -408,7 +408,7 @@
 diff --git a/stdlib/.depend b/stdlib/.depend
 --- a/stdlib/.depend
 +++ b/stdlib/.depend
-@@ -400,15 +400,18 @@ stdlib__In_channel.cmo : in_channel.ml \
+@@ -356,15 +356,18 @@ stdlib__In_channel.cmo : in_channel.ml \
      stdlib.cmi \
      stdlib__Fun.cmi \
      stdlib__Bytes.cmi \
@@ -428,7 +428,7 @@
  stdlib__Int.cmo : int.ml \
      stdlib.cmi \
      stdlib__Int.cmi
-@@ -556,13 +559,16 @@ stdlib__Option.cmi : option.mli \
+@@ -507,13 +510,16 @@ stdlib__Option.cmi : option.mli \
  stdlib__Out_channel.cmo : out_channel.ml \
      stdlib.cmi \
      stdlib__Fun.cmi \
