--- a	2023-09-28 18:43:26.403049274 +0100
+++ b	2023-09-28 18:43:26.411049610 +0100
@@ -1,22 +1,3 @@
-diff --git a/stdlib/.depend b/stdlib/.depend
---- a/stdlib/.depend
-+++ b/stdlib/.depend
-@@ -217,6 +217,7 @@ stdlib__Domain.cmo : domain.ml \
-     stdlib.cmi \
-     stdlib__Obj.cmi \
-     stdlib__Mutex.cmi \
-+    stdlib__List.cmi \
-     stdlib__Atomic.cmi \
-     stdlib__Array.cmi \
-     stdlib__Domain.cmi
-@@ -225,6 +226,7 @@ stdlib__Domain.cmx : domain.ml \
-     stdlib.cmx \
-     stdlib__Obj.cmx \
-     stdlib__Mutex.cmx \
-+    stdlib__List.cmx \
-     stdlib__Atomic.cmx \
-     stdlib__Array.cmx \
-     stdlib__Domain.cmi
 diff --git a/stdlib/domain.ml b/stdlib/domain.ml
 --- a/stdlib/domain.ml
 +++ b/stdlib/domain.ml
@@ -128,16 +109,16 @@
 diff --git a/stdlib/filename.ml b/stdlib/filename.ml
 --- a/stdlib/filename.ml
 +++ b/stdlib/filename.ml
-@@ -326,7 +326,8 @@ let remove_extension name =
- external open_desc: string -> open_flag list -> int -> int = "caml_sys_open"
+@@ -327,7 +327,8 @@ external open_desc: string -> open_flag list -> int -> int = "caml_sys_open"
  external close_desc: int -> unit = "caml_sys_close"
  
+ (* BACKPORT BEGIN
 -let prng_key = Domain.DLS.new_key Random.State.make_self_init
 +let prng_key =
 +  Domain.DLS.new_key Random.State.make_self_init
- 
- let temp_file_name temp_dir prefix suffix =
-   let random_state = Domain.DLS.get prng_key in
+ *)
+ let prng_key = lazy(Random.State.make_self_init ())
+ (* BACKPORT END *)
 diff --git a/stdlib/random.ml b/stdlib/random.ml
 --- a/stdlib/random.ml
 +++ b/stdlib/random.ml
@@ -151,10 +132,10 @@
  end
  
  (* This is the state you get with [init 27182818] and then applying
-@@ -212,7 +215,8 @@ let mk_default () = {
-   State.idx = 0;
+@@ -213,7 +216,8 @@ let mk_default () = {
  }
  
+ (* BACKPORT BEGIN
 -let random_key = Domain.DLS.new_key mk_default
 +let random_key =
 +  Domain.DLS.new_key ~split_from_parent:State.split mk_default
@@ -183,7 +164,7 @@
 diff --git a/testsuite/tests/gc-roots/globroots.ml b/testsuite/tests/gc-roots/globroots.ml
 --- a/testsuite/tests/gc-roots/globroots.ml
 +++ b/testsuite/tests/gc-roots/globroots.ml
-@@ -30,7 +30,10 @@ module Test(G: GLOBREF) () = struct
+@@ -32,7 +32,10 @@ module Test(G: GLOBREF) () = struct
  
    let size = 1024
  
@@ -262,7 +243,7 @@
 diff --git a/testsuite/tests/weak-ephe-final/weaklifetime_par.ml b/testsuite/tests/weak-ephe-final/weaklifetime_par.ml
 --- a/testsuite/tests/weak-ephe-final/weaklifetime_par.ml
 +++ b/testsuite/tests/weak-ephe-final/weaklifetime_par.ml
-@@ -3,7 +3,10 @@
+@@ -5,7 +5,10 @@
  
  let size = 1000;;
  let num_domains = 4;;
