--- a	2023-09-28 18:45:44.896612196 +0100
+++ b	2023-09-28 18:45:44.904612504 +0100
@@ -1,10 +1,10 @@
 diff --git a/otherlibs/systhreads/thread.ml b/otherlibs/systhreads/thread.ml
 --- a/otherlibs/systhreads/thread.ml
 +++ b/otherlibs/systhreads/thread.ml
-@@ -88,6 +88,7 @@ let () =
-   thread_initialize ();
+@@ -106,6 +106,7 @@ let () =
    Domain.at_each_spawn thread_initialize_domain;
    Sys.set_signal preempt_signal (Sys.Signal_handle preempt);
+ *)
 +  (* Callback in [caml_shutdown], when the last domain exits. *)
    Callback.register "Thread.at_shutdown" (fun () ->
      thread_cleanup();
