--- a	2023-09-28 18:47:04.031593263 +0100
+++ b	2023-09-28 18:47:04.055594148 +0100
@@ -14,7 +14,7 @@
 diff --git a/stdlib/.depend b/stdlib/.depend
 --- a/stdlib/.depend
 +++ b/stdlib/.depend
-@@ -358,6 +358,7 @@ stdlib__Gc.cmo : gc.ml \
+@@ -314,6 +314,7 @@ stdlib__Gc.cmo : gc.ml \
      stdlib__String.cmi \
      stdlib__Printf.cmi \
      stdlib__Printexc.cmi \
@@ -22,7 +22,7 @@
      stdlib__Atomic.cmi \
      stdlib__Gc.cmi
  stdlib__Gc.cmx : gc.ml \
-@@ -365,6 +366,7 @@ stdlib__Gc.cmx : gc.ml \
+@@ -321,6 +322,7 @@ stdlib__Gc.cmx : gc.ml \
      stdlib__String.cmx \
      stdlib__Printf.cmx \
      stdlib__Printexc.cmx \
@@ -33,7 +33,7 @@
 diff --git a/stdlib/gc.ml b/stdlib/gc.ml
 --- a/stdlib/gc.ml
 +++ b/stdlib/gc.ml
-@@ -108,8 +108,8 @@ type alarm_rec = {active : alarm; f : unit -> unit}
+@@ -114,8 +114,8 @@ type alarm_rec = {active : alarm; f : unit -> unit}
  
  let rec call_alarm arec =
    if Atomic.get arec.active then begin
