--- a	2023-09-28 18:43:41.931697959 +0100
+++ b	2023-09-28 18:43:41.939698291 +0100
@@ -14,7 +14,7 @@
 diff --git a/stdlib/Makefile b/stdlib/Makefile
 --- a/stdlib/Makefile
 +++ b/stdlib/Makefile
-@@ -258,13 +258,17 @@ include .depend
+@@ -258,6 +258,10 @@ include .depend
  
  STDLIB_NAMESPACE_MODULES = $(subst $(SPACE),|,$(STDLIB_PREFIXED_MODULES))
  
@@ -24,7 +24,8 @@
 +
  .PHONY: depend
  depend:
- 	$(CAMLDEP) $(DEPFLAGS) $(filter-out stdlib.%,$(wildcard *.mli *.ml)) \
+ 	$(CAMLDEP) $(DEPFLAGS) $(filter-out stdlib.%,std_exit.ml \
+@@ -265,7 +269,7 @@ depend:
  	  > .depend.tmp
  	$(CAMLDEP) $(DEPFLAGS) -pp "$(AWK) -f ./remove_module_aliases.awk" \
  	  stdlib.ml stdlib.mli >> .depend.tmp
