--- a	2023-09-28 18:48:18.346284801 +0100
+++ b	2023-09-28 18:48:18.378285939 +0100
@@ -1,19 +1,16 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -237,6 +237,10 @@ Working version
+@@ -237,6 +237,9 @@ Working version
    (Vincent Laviron, report by François Pottier, review by Nathanaëlle Courant
    and Gabriel Scherer)
  
 +- #12334, #12368: Bad error message with mutually recursive abbreviations
 +  (Jacques Garrigue, report by Richard Eisenberg, review by Gabriel Scherer
 +   and Richard Eisenberg)
-+
+ 
  - #12401: `seek_in` and `seek_out` sometimes returned normally when given
    negative offsets, instead of failing. Now both functions should consistently
-   raise `Sys_error` in this case.
-diff --git a/boot/ocamlc b/boot/ocamlc
-Binary files a/boot/ocamlc and b/boot/ocamlc differ
 diff --git a/ocamldoc/odoc_sig.ml b/ocamldoc/odoc_sig.ml
 --- a/ocamldoc/odoc_sig.ml
 +++ b/ocamldoc/odoc_sig.ml
