--- a	2023-09-28 18:44:42.214154061 +0100
+++ b	2023-09-28 18:44:42.234154860 +0100
@@ -48,9 +48,9 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -950,6 +950,12 @@ AS_IF([! $arch64],
-   [AS_CASE([$ocaml_cv_cc_vendor],
-     [gcc-*], [cclibs="$cclibs -latomic"])])
+@@ -962,6 +962,12 @@ dnl AS_IF([! $arch64],
+ dnl  [AS_CASE([$ocaml_cv_cc_vendor],
+ dnl    [gcc-*], [cclibs="$cclibs -latomic"])])
  
 +# Support for C11 atomic types
 +
