--- a	2023-09-28 18:47:20.436195842 +0100
+++ b	2023-09-28 18:47:20.448196281 +0100
@@ -31,7 +31,7 @@
  ocaml.tmp: $(ocaml_LIBRARIES:=.cma) $(ocaml_MODULES:=.cmo)
  	$(V_LINKC)$(LINK_BYTECODE_PROGRAM) -o $@ $^
  
-@@ -912,22 +912,24 @@ runtime/%.bi.$(O): OC_CPPFLAGS += $(ocamlruni_CPPFLAGS)
+@@ -909,22 +909,24 @@ runtime/%.bi.$(O): OC_CPPFLAGS += $(ocamlruni_CPPFLAGS)
  $(DEPDIR)/runtime/%.bi.$(D): OC_CPPFLAGS += $(ocamlruni_CPPFLAGS)
  
  runtime/%.bpic.$(O): OC_CFLAGS += $(SHAREDLIB_CFLAGS)
@@ -60,7 +60,7 @@
  
  ## Compilation of runtime C files
  
-@@ -1120,7 +1122,7 @@ ocamllex: ocamlyacc
+@@ -1119,7 +1121,7 @@ ocamllex: ocamlyacc
  ocamllex.opt: ocamlopt
  	$(MAKE) lex-allopt
  
@@ -315,8 +315,8 @@
  AC_SUBST([afl])
  AC_SUBST([flexlink_cmd])
  AC_SUBST([flexdll_chain])
-@@ -1273,6 +1275,9 @@ AS_CASE([$host],
-     [has_native_backend=yes; arch=riscv; model=riscv64; system=linux]
+@@ -1287,6 +1289,9 @@ AS_CASE([$host],
+     [arch=riscv; model=riscv64; system=linux]
  )
  
 +native_cflags=''
@@ -325,7 +325,7 @@
  AS_CASE([$ccomptype],
    [msvc],
      [runtime_asm_objects=${arch}nt.${OBJEXT}],
-@@ -2143,7 +2148,7 @@ AS_IF([test x"$enable_mmap_map_stack" = "xyes"],
+@@ -2193,7 +2198,7 @@ AS_IF([test x"$enable_mmap_map_stack" = "xyes"],
       [with_mmap_map_stack=false])
    ])
  
@@ -334,7 +334,7 @@
  
  AS_IF([test x"$enable_function_sections" = "xno"],
    [function_sections=false],
-@@ -2165,7 +2170,7 @@ AS_IF([test x"$enable_function_sections" = "xno"],
+@@ -2215,7 +2220,7 @@ AS_IF([test x"$enable_function_sections" = "xno"],
                in Clang prior to version 3.5.])],
              [gcc-*|clang-*],
                [function_sections=true;
