--- a	2023-09-28 18:46:19.265922626 +0100
+++ b	2023-09-28 18:46:19.281923230 +0100
@@ -1,7 +1,7 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -807,7 +807,8 @@ OCaml 5.0
+@@ -802,7 +802,8 @@ OCaml 5.0
  
  - #11533, #11534: follow synonyms again in #show_module_type
    (this had stopped working in 4.14.0)
@@ -9,5 +9,5 @@
 +  (Gabriel Scherer, review by Jacques Garrigue and Florian Angeletti,
 +   report by Yaron Minsky)
  
- OCaml 4.14.0 (28 March 2022)
- ----------------------------
+ OCaml 4.14 maintenance branch
+ -----------------------------
