--- a	2023-09-28 18:45:37.448324841 +0100
+++ b	2023-09-28 18:45:37.456325150 +0100
@@ -1,7 +1,3 @@
-diff --git a/boot/ocamlc b/boot/ocamlc
-Binary files a/boot/ocamlc and b/boot/ocamlc differ
-diff --git a/boot/ocamllex b/boot/ocamllex
-Binary files a/boot/ocamllex and b/boot/ocamllex differ
 diff --git a/manual/src/refman/extensions/effects.etex b/manual/src/refman/extensions/effects.etex
 --- a/manual/src/refman/extensions/effects.etex
 +++ b/manual/src/refman/extensions/effects.etex
@@ -17,7 +13,7 @@
 +handle the effect "F". Hence, we will get an "Effect.Unhandled F" exception
 +when we run "bar".
  
- \begin{caml_example}{verbatim}
+ \begin{caml_example}{verbatim}[error]
 -try bar () with Unhandled -> "Saw unhandled exception"
 +try bar () with Effect.Unhandled F -> "Saw Effect.Unhandled exception"
  \end{caml_example}
@@ -30,29 +26,25 @@
 -"Continuation_already_taken" exception. For example:
 +"Continuation_already_resumed" exception. For example:
  
- \begin{caml_example}{verbatim}
+ \begin{caml_example}{verbatim}[error]
  try_with perform (Xchg 0)
 diff --git a/stdlib/.depend b/stdlib/.depend
 --- a/stdlib/.depend
 +++ b/stdlib/.depend
-@@ -229,10 +229,16 @@ stdlib__Domain.cmx : domain.ml \
-     stdlib__Domain.cmi
- stdlib__Domain.cmi : domain.mli
- stdlib__Effect.cmo : effect.ml \
-+    stdlib__Printf.cmi \
-     stdlib__Printexc.cmi \
-+    stdlib__Obj.cmi \
-+    stdlib__Callback.cmi \
-     stdlib__Effect.cmi
- stdlib__Effect.cmx : effect.ml \
-+    stdlib__Printf.cmx \
-     stdlib__Printexc.cmx \
-+    stdlib__Obj.cmx \
-+    stdlib__Callback.cmx \
-     stdlib__Effect.cmi
- stdlib__Effect.cmi : effect.mli \
-     stdlib__Printexc.cmi
-@@ -580,7 +586,8 @@ stdlib__Printexc.cmx : printexc.ml \
+@@ -410,12 +410,10 @@ stdlib__Lexing.cmx : lexing.ml \
+     stdlib__Lexing.cmi
+ stdlib__Lexing.cmi : lexing.mli
+ stdlib__List.cmo : list.ml \
+-    stdlib__Sys.cmi \
+     stdlib__Seq.cmi \
+     stdlib__Either.cmi \
+     stdlib__List.cmi
+ stdlib__List.cmx : list.ml \
+-    stdlib__Sys.cmx \
+     stdlib__Seq.cmx \
+     stdlib__Either.cmx \
+     stdlib__List.cmi
+@@ -537,7 +535,8 @@ stdlib__Printexc.cmx : printexc.ml \
      stdlib__Atomic.cmx \
      stdlib__Array.cmx \
      stdlib__Printexc.cmi
@@ -62,20 +54,6 @@
  stdlib__Printf.cmo : printf.ml \
      camlinternalFormatBasics.cmi \
      camlinternalFormat.cmi \
-diff --git a/stdlib/StdlibModules b/stdlib/StdlibModules
---- a/stdlib/StdlibModules
-+++ b/stdlib/StdlibModules
-@@ -40,8 +40,8 @@ STDLIB_MODULE_BASENAMES = \
-   list int bytes string unit marshal array float int32 int64 nativeint \
-   lexing parsing set map stack queue buffer \
-   mutex condition semaphore domain \
--  camlinternalFormat printf arg \
--  printexc fun gc digest bigarray random hashtbl weak \
-+  camlinternalFormat printf arg printexc \
-+  fun gc digest bigarray random hashtbl weak \
-   format scanf callback camlinternalOO oo camlinternalMod ephemeron \
-   filename complex arrayLabels listLabels bytesLabels stringLabels moreLabels \
-   stdLabels in_channel out_channel effect
 diff --git a/stdlib/effect.ml b/stdlib/effect.ml
 --- a/stdlib/effect.ml
 +++ b/stdlib/effect.ml
@@ -254,7 +232,7 @@
 diff --git a/testsuite/tests/callback/test7.ml b/testsuite/tests/callback/test7.ml
 --- a/testsuite/tests/callback/test7.ml
 +++ b/testsuite/tests/callback/test7.ml
-@@ -31,8 +31,8 @@ let _ =
+@@ -33,8 +33,8 @@ let _ =
      printf "[Caml] Call caml_to_c\n%!";
      begin try
        caml_to_c ()
@@ -307,7 +285,7 @@
 diff --git a/testsuite/tests/effects/test6.ml b/testsuite/tests/effects/test6.ml
 --- a/testsuite/tests/effects/test6.ml
 +++ b/testsuite/tests/effects/test6.ml
-@@ -8,15 +8,23 @@ type _ t += E : unit t
+@@ -10,15 +10,23 @@ type _ t += E : unit t
            | F : unit t
  
  let () =
