--- a	2023-09-28 18:43:40.963657720 +0100
+++ b	2023-09-28 18:43:40.979658386 +0100
@@ -53,7 +53,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -389,6 +389,10 @@ AC_ARG_ENABLE([function-sections],
+@@ -398,6 +398,10 @@ AC_ARG_ENABLE([function-sections],
    [],
    [enable_function_sections=auto])
  
@@ -64,7 +64,7 @@
  AC_ARG_WITH([afl],
    [AS_HELP_STRING([--with-afl],
      [use the AFL fuzzer])])
-@@ -1885,6 +1889,25 @@ AS_IF([test x"$enable_flat_float_array" = "xno"],
+@@ -1919,6 +1923,25 @@ AS_IF([test x"$enable_flat_float_array" = "xno"],
    [AC_DEFINE([FLAT_FLOAT_ARRAY])
    flat_float_array=true])
  
