--- a	2023-09-28 18:45:31.464093075 +0100
+++ b	2023-09-28 18:45:31.484093851 +0100
@@ -1,15 +1,15 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -607,6 +607,9 @@ OCaml 5.0
-    review by Xavier Leroy,
-    report by Jan Midtgaard and Tom Kelly)
+@@ -602,6 +602,9 @@ OCaml 5.0
+   mingw-w64 i686 port.
+   (David Allsopp, review by Xavier Leroy and Sébastien Hinderer)
  
 +- #11516, #11524: Fix the `deprecated_mutable` attribute.
 +  (Chris Casinghino, review by Nicolás Ojeda Bär and Florian Angeletti)
 +
- OCaml 4.14.0 (28 March 2022)
- ----------------------------
+ OCaml 4.14 maintenance branch
+ -----------------------------
  
 diff --git a/testsuite/tests/warnings/deprecated_mutable.compilers.reference b/testsuite/tests/warnings/deprecated_mutable.compilers.reference
 new file mode 100644
