--- a	2023-09-28 18:44:57.162748713 +0100
+++ b	2023-09-28 18:44:57.166748872 +0100
@@ -1,7 +1,7 @@
 diff --git a/testsuite/tests/parallel/domain_parallel_spawn_burn.ml b/testsuite/tests/parallel/domain_parallel_spawn_burn.ml
 --- a/testsuite/tests/parallel/domain_parallel_spawn_burn.ml
 +++ b/testsuite/tests/parallel/domain_parallel_spawn_burn.ml
-@@ -10,6 +10,11 @@ let test_size =
+@@ -12,6 +12,11 @@ let test_size =
    try int_of_string (Sys.getenv "OCAML_TEST_SIZE")
    with Not_found | Failure _ -> 0
  
@@ -13,7 +13,7 @@
  let (list_size, num_domains) =
    if test_size >= 2 then (14, 25) else (13, 12)
  
-@@ -21,8 +26,7 @@ let rec burn l =
+@@ -23,8 +28,7 @@ let rec burn l =
  let test_parallel_spawn () =
    for i = 1 to 20 do
      Array.init num_domains (fun _ -> Domain.spawn (fun () -> burn [0]))
@@ -23,7 +23,7 @@
    done
  
  let () =
-@@ -40,8 +44,6 @@ let () =
+@@ -42,8 +46,6 @@ let () =
  
    running := false;
    join domain_minor_gc;
