--- a	2023-09-28 18:46:57.259343030 +0100
+++ b	2023-09-28 18:46:57.275343622 +0100
@@ -75,7 +75,7 @@
  endif
  	$(MAKE) -C stdlib install
  ifeq "$(INSTALL_BYTECODE_PROGRAMS)" "true"
-@@ -1735,7 +1738,8 @@ ifneq "$(runtime_NATIVE_SHARED_LIBRARIES)" ""
+@@ -1738,7 +1741,8 @@ ifneq "$(runtime_NATIVE_SHARED_LIBRARIES)" ""
  	$(INSTALL_PROG) $(runtime_NATIVE_SHARED_LIBRARIES) "$(INSTALL_LIBDIR)"
  endif
  ifeq "$(INSTALL_BYTECODE_PROGRAMS)" "true"
