--- a	2023-09-28 18:46:48.295010455 +0100
+++ b	2023-09-28 18:46:48.311011050 +0100
@@ -15,7 +15,7 @@
 diff --git a/lambda/matching.ml b/lambda/matching.ml
 --- a/lambda/matching.ml
 +++ b/lambda/matching.ml
-@@ -2376,11 +2376,29 @@ module SArg = struct
+@@ -2424,11 +2424,29 @@ module SArg = struct
    let make_if cond ifso ifnot = Lifthenelse (cond, ifso, ifnot)
  
    let make_switch loc arg cases acts =
@@ -46,7 +46,7 @@
        ( arg,
          { sw_numconsts = Array.length cases;
            sw_consts = !l;
-@@ -2388,7 +2406,7 @@ module SArg = struct
+@@ -2436,7 +2454,7 @@ module SArg = struct
            sw_blocks = [];
            sw_failaction = None
          },
