--- a	2023-09-28 18:46:18.305886359 +0100
+++ b	2023-09-28 18:46:18.325887115 +0100
@@ -1,7 +1,7 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -799,6 +799,10 @@ OCaml 5.0
+@@ -794,6 +794,10 @@ OCaml 5.0
    multiple threads.
    (Marc Lasson, Nicolás Ojeda Bär, review by Gabriel Scherer and David Allsopp)
  
@@ -9,8 +9,8 @@
 +  (this had stopped working in 4.14.0)
 +  (Gabriel Scherer, review by Jacques Garrigue, report by Yaron Minsky)
 +
- OCaml 4.14.0 (28 March 2022)
- ----------------------------
+ OCaml 4.14 maintenance branch
+ -----------------------------
  
 diff --git a/testsuite/tests/tool-toplevel/show.ml b/testsuite/tests/tool-toplevel/show.ml
 --- a/testsuite/tests/tool-toplevel/show.ml
