--- a	2023-09-28 18:47:11.191856890 +0100
+++ b	2023-09-28 18:47:11.199857184 +0100
@@ -1,7 +1,7 @@
 diff --git a/testsuite/tests/lib-marshal/intext_par.ml b/testsuite/tests/lib-marshal/intext_par.ml
 --- a/testsuite/tests/lib-marshal/intext_par.ml
 +++ b/testsuite/tests/lib-marshal/intext_par.ml
-@@ -482,7 +482,7 @@ let test_objects () =
+@@ -484,7 +484,7 @@ let test_objects () =
    test 509 (x#test2 = false);
    test 510 (x#test3 = "footest5test3test442");
    test 511 (x#test4 = 42L);
