--- a	2023-09-28 18:47:41.428959819 +0100
+++ b	2023-09-28 18:47:41.448960543 +0100
@@ -295,17 +295,3 @@
 -
 -let fundecl f = (new scheduler)#schedule_fundecl f
 +let fundecl f = f
-diff --git a/configure b/configure
-Binary files a/configure and b/configure differ
-diff --git a/configure.ac b/configure.ac
---- a/configure.ac
-+++ b/configure.ac
-@@ -1203,7 +1203,7 @@ AS_CASE([$host],
-   [[powerpc*-*-linux*]],
-     [arch=power; AS_IF([$arch64],[model=ppc64],[model=ppc]); system=elf],
-   [[s390x*-*-linux*]],
--    [arch=s390x; model=z10; system=elf],
-+    [has_native_backend=yes; arch=s390x; model=z10; system=elf],
-   # expected to match "gnueabihf" as well as "musleabihf"
-   [armv6*-*-linux-*eabihf],
-     [arch=arm; model=armv6; system=linux_eabihf],
