--- a	2023-09-28 18:47:33.700679483 +0100
+++ b	2023-09-28 18:47:33.716680065 +0100
@@ -19,10 +19,10 @@
 diff --git a/asmcomp/amd64/emit.mlp b/asmcomp/amd64/emit.mlp
 --- a/asmcomp/amd64/emit.mlp
 +++ b/asmcomp/amd64/emit.mlp
-@@ -537,13 +537,8 @@ let emit_instr env fallthrough i =
-       end
-   | Lop(Iextcall { func; alloc; stack_ofs }) ->
+@@ -542,13 +542,8 @@ let emit_instr env fallthrough i =
+   | Lop(Iextcall { func; alloc; stack_ofs = _ }) ->
        add_used_symbol func;
+ (* BACKPORT
 -      let base_stack_size =
 -        if Arch.win64 then
 -          32 (* Windows x64 rcx+rdx+r8+r9 shadow stack *)
@@ -47,19 +47,23 @@
    for i = 0 to Array.length arg - 1 do
      match arg.(i) with
      | Val | Int | Addr as ty ->
-@@ -291,12 +291,12 @@ let destroyed_at_c_call =
+@@ -291,16 +291,16 @@ let destroyed_at_c_call =
       by the code sequence used for C calls in emit.mlp, so it
       is marked as destroyed. *)
    if win64 then
 -    (* Win64: rbx, rsi, rdi, r12-r15, xmm6-xmm15 preserved *)
 +    (* Win64: rsi, rdi, r12-r15, xmm6-xmm15 preserved *)
      Array.of_list(List.map phys_reg
+ (* BACKPORT BEGIN
 -      [0;4;5;6;7;10;11;12;
-+      [0;1;4;5;6;7;10;11;12;
++      [0;1;4;5;6;7;10;11;12
+ *)
+       [0;4;5;6;7;10;11;
+ (* BACKPORT END *)
         100;101;102;103;104;105])
    else
 -    (* Unix: rbx, r12-r15 preserved *)
 +    (* Unix: r12-r15 preserved *)
      Array.of_list(List.map phys_reg
+ (* BACKPORT BEGIN
        [0;1;2;3;4;5;6;7;10;11;
-        100;101;102;103;104;105;106;107;
