--- a	2023-09-28 18:47:42.456997031 +0100
+++ b	2023-09-28 18:47:42.460997176 +0100
@@ -1,7 +1,7 @@
 diff --git a/.gitattributes b/.gitattributes
 --- a/.gitattributes
 +++ b/.gitattributes
-@@ -175,7 +175,7 @@ build-aux/install text eol=lf
+@@ -176,7 +176,7 @@ build-aux/install text eol=lf
  build-aux/missing text eol=lf
  ocamldoc/remove_DEBUG text eol=lf
  ocamltest/getocamloptdefaultflags text eol=lf
