--- a	2023-09-28 18:45:47.336706066 +0100
+++ b	2023-09-28 18:45:47.352706682 +0100
@@ -14,7 +14,7 @@
 diff --git a/stdlib/.depend b/stdlib/.depend
 --- a/stdlib/.depend
 +++ b/stdlib/.depend
-@@ -322,6 +322,7 @@ stdlib__Format.cmo : format.ml \
+@@ -279,6 +279,7 @@ stdlib__Format.cmo : format.ml \
      camlinternalFormat.cmi \
      stdlib__Bytes.cmi \
      stdlib__Buffer.cmi \
@@ -22,7 +22,7 @@
      stdlib__Format.cmi
  stdlib__Format.cmx : format.ml \
      stdlib__String.cmx \
-@@ -337,6 +338,7 @@ stdlib__Format.cmx : format.ml \
+@@ -294,6 +295,7 @@ stdlib__Format.cmx : format.ml \
      camlinternalFormat.cmx \
      stdlib__Bytes.cmx \
      stdlib__Buffer.cmx \
@@ -33,7 +33,7 @@
 diff --git a/stdlib/format.ml b/stdlib/format.ml
 --- a/stdlib/format.ml
 +++ b/stdlib/format.ml
-@@ -1250,30 +1250,25 @@ and set_tags v =
+@@ -1264,30 +1264,25 @@ and set_tags v =
  
  (* Convenience functions *)
  
@@ -82,7 +82,7 @@
 diff --git a/stdlib/format.mli b/stdlib/format.mli
 --- a/stdlib/format.mli
 +++ b/stdlib/format.mli
-@@ -1152,17 +1152,43 @@ val formatter_of_symbolic_output_buffer : symbolic_output_buffer -> formatter
+@@ -1155,17 +1155,43 @@ val formatter_of_symbolic_output_buffer : symbolic_output_buffer -> formatter
  
  (** {1 Convenience formatting functions.} *)
  
