--- a	2023-09-28 18:44:28.877618853 +0100
+++ b	2023-09-28 18:44:28.885619176 +0100
@@ -68,8 +68,8 @@
  AC_SUBST([fpic])
  AC_SUBST([mkexe])
  AC_SUBST([mkexedebugflag])
-@@ -1127,6 +1126,11 @@ dnl [riscv64-*-linux*],
- dnl   [arch=riscv; model=riscv64; system=linux]
+@@ -1137,6 +1136,11 @@ AS_CASE([$host],
+     [arch=riscv; model=riscv64; system=linux]
  )
  
 +AS_CASE([$ccomptype],
@@ -80,7 +80,7 @@
  AS_CASE([$enable_native_compiler,$arch],
    [no,*],
      [native_compiler=false
-@@ -1970,6 +1974,12 @@ AS_IF([test x"$with_afl" = "xyes"],
+@@ -2005,6 +2009,12 @@ AS_IF([test x"$with_afl" = "xyes"],
    [afl=true],
    [afl=false])
  
