--- a	2023-09-28 18:48:01.489682661 +0100
+++ b	2023-09-28 18:48:01.509683378 +0100
@@ -29,7 +29,7 @@
      | Lop(Ialloc _) -> 5
      | Lop(Ispecific(Ialloc_far _)) -> 6
      | Lop(Ipoll { return_label = Some(_) }) -> 5
-@@ -678,6 +683,10 @@ let emit_instr env i =
+@@ -680,6 +685,10 @@ let emit_instr env i =
          end;
          if memory_chunk = Byte_signed then
            `	extsb	{emit_reg i.res.(0)}, {emit_reg i.res.(0)}\n`
@@ -40,7 +40,7 @@
      | Lop(Istore(chunk, addr, assignment)) ->
          let storeinstr =
            match chunk with
-@@ -685,7 +694,7 @@ let emit_instr env i =
+@@ -687,7 +696,7 @@ let emit_instr env i =
            | Sixteen_unsigned | Sixteen_signed -> "sth"
            | Thirtytwo_unsigned | Thirtytwo_signed -> "stw"
            | Word_int | Word_val -> "std"
