--- a	2023-09-28 18:48:22.458430983 +0100
+++ b	2023-09-28 18:48:22.466431268 +0100
@@ -1,7 +1,7 @@
 diff --git a/testsuite/tests/lib-marshal/intext_par.ml b/testsuite/tests/lib-marshal/intext_par.ml
 --- a/testsuite/tests/lib-marshal/intext_par.ml
 +++ b/testsuite/tests/lib-marshal/intext_par.ml
-@@ -11,6 +11,12 @@
+@@ -13,6 +13,12 @@
  
  (* Test for output_value / input_value *)
  
@@ -14,7 +14,7 @@
  let max_data_depth = 500000
  
  type t = A | B of int | C of float | D of string | E of char
-@@ -595,7 +601,7 @@ let main_domain id =
+@@ -597,7 +603,7 @@ let main_domain id =
    ()
  
  let main () =
