--- a	2023-09-28 18:47:00.323456355 +0100
+++ b	2023-09-28 18:47:00.351457390 +0100
@@ -33,7 +33,7 @@
 diff --git a/testsuite/tests/lib-runtime-events/test_user_event_unknown.ml b/testsuite/tests/lib-runtime-events/test_user_event_unknown.ml
 --- a/testsuite/tests/lib-runtime-events/test_user_event_unknown.ml
 +++ b/testsuite/tests/lib-runtime-events/test_user_event_unknown.ml
-@@ -13,6 +13,9 @@ type User.tag += Ev of int
+@@ -15,6 +15,9 @@ type User.tag += Ev of int
  let ev0 =
    User.register "ev0" (Ev 0) Type.int
  
@@ -43,7 +43,7 @@
  let () =
    start ();
    let parent_cwd = Sys.getcwd () in
-@@ -23,11 +26,14 @@ let () =
+@@ -25,11 +28,14 @@ let () =
      let ev1 = User.register "ev1" (Ev 1) Type.int in
      let ev2 = User.register "ev2" (Ev 2) Type.int in
      let ev3 = User.register "ev3" (Ev 3) Type.span in
