--- a	2023-09-28 18:45:28.567980620 +0100
+++ b	2023-09-28 18:45:28.587981397 +0100
@@ -1,18 +1,18 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -587,6 +587,11 @@ OCaml 5.0
-   mingw-w64 i686 port.
-   (David Allsopp, review by Xavier Leroy and Sébastien Hinderer)
+@@ -590,6 +590,11 @@ OCaml 5.0
+ OCaml 4.14 maintenance branch
+ -----------------------------
  
 +- #11508, #11509: make Bytes.escaped domain-safe
 +  (Christiano Haesbaert and Gabriel Scherer,
 +   review by Xavier Leroy,
 +   report by Jan Midtgaard and Tom Kelly)
 +
- OCaml 4.14.0 (28 March 2022)
- ----------------------------
+ ### Manual and documentation:
  
+ - #11045, #11409: document that the array argument to `caml_callbackN`
 diff --git a/stdlib/bytes.ml b/stdlib/bytes.ml
 --- a/stdlib/bytes.ml
 +++ b/stdlib/bytes.ml
