--- a	2023-09-28 18:47:44.461069519 +0100
+++ b	2023-09-28 18:47:44.477070098 +0100
@@ -22,7 +22,7 @@
 diff --git a/testsuite/tests/parallel/major_gc_wait_backup.ml b/testsuite/tests/parallel/major_gc_wait_backup.ml
 --- a/testsuite/tests/parallel/major_gc_wait_backup.ml
 +++ b/testsuite/tests/parallel/major_gc_wait_backup.ml
-@@ -37,6 +37,6 @@ let _ =
+@@ -39,6 +39,6 @@ let _ =
    ) in
    Gc.full_major ();
    let n = major_collections () in
