--- a	2023-09-28 18:47:30.456561490 +0100
+++ b	2023-09-28 18:47:30.468561927 +0100
@@ -65,34 +65,6 @@
  |=====
  
  = README =
-diff --git a/VERSION b/VERSION
---- a/VERSION
-+++ b/VERSION
-@@ -1,4 +1,4 @@
--5.1.0+dev2-2023-04-11
-+5.2.0+dev0-2023-04-11
- 
- # Starting with OCaml 4.14, although the version string that appears above is
- # still correct and this file can thus still be used to figure it out,
-diff --git a/build-aux/ocaml_version.m4 b/build-aux/ocaml_version.m4
---- a/build-aux/ocaml_version.m4
-+++ b/build-aux/ocaml_version.m4
-@@ -32,11 +32,11 @@ m4_define([OCAML__DEVELOPMENT_VERSION], [true])
- # including the patchlevel, are mandatory.
- 
- m4_define([OCAML__VERSION_MAJOR], [5])
--m4_define([OCAML__VERSION_MINOR], [1])
-+m4_define([OCAML__VERSION_MINOR], [2])
- m4_define([OCAML__VERSION_PATCHLEVEL], [0])
- # Note that the OCAML__VERSION_EXTRA string defined below is always empty
- # for officially-released versions of OCaml.
--m4_define([OCAML__VERSION_EXTRA], [dev2-2023-04-11])
-+m4_define([OCAML__VERSION_EXTRA], [dev0-2023-04-11])
- 
- # The OCAML__VERSION_EXTRA_PREFIX macro defined below should be a
- # single character:
-diff --git a/configure b/configure
-Binary files a/configure and b/configure differ
 diff --git a/ocaml-variants.opam b/ocaml-variants.opam
 --- a/ocaml-variants.opam
 +++ b/ocaml-variants.opam
