--- a	2023-09-28 18:48:30.466714888 +0100
+++ b	2023-09-28 18:48:30.490715737 +0100
@@ -28,7 +28,7 @@
    result in wrong behavior, but it should preserve memory-safety.
    (Florian Angeletti and Gabriel Scherer, review by Gabriel Scherer
     and Vincent Laviron, report by David Allsopp)
-@@ -2921,7 +2921,7 @@ OCaml 4.13.0 (24 September 2021)
+@@ -2956,7 +2956,7 @@ OCaml 4.13.0 (24 September 2021)
    (Xavier Leroy, report by Richard Bornat, review by Stephen Dolan
     and Mark Shinwell)
  
@@ -37,7 +37,7 @@
    `-output-complete-exe`.
    (Nicolás Ojeda Bär, review by David Allsopp)
  
-@@ -4548,7 +4548,7 @@ OCaml 4.10.0 (21 February 2020)
+@@ -4583,7 +4583,7 @@ OCaml 4.10.0 (21 February 2020)
    (Florian Angeletti, report by Anton Kochkov, review by Gabriel Scherer)
  
  - #9101: add links to section anchor before the section title,
