--- a	2023-09-28 18:45:09.947252994 +0100
+++ b	2023-09-28 18:45:09.967253780 +0100
@@ -8,8 +8,8 @@
 +- #11417: Fix regression allowing virtual methods in non-virtual classes.
 +  (Leo White, review by Florian Angeletti)
 +
- OCaml 4.14.0 (28 March 2022)
- ----------------------------
+ OCaml 4.14 maintenance branch
+ -----------------------------
  
 diff --git a/testsuite/tests/typing-objects/Tests.ml b/testsuite/tests/typing-objects/Tests.ml
 --- a/testsuite/tests/typing-objects/Tests.ml
