--- a	2023-09-28 18:44:18.445197031 +0100
+++ b	2023-09-28 18:44:18.465197843 +0100
@@ -154,7 +154,7 @@
 diff --git a/stdlib/hashtbl.ml b/stdlib/hashtbl.ml
 --- a/stdlib/hashtbl.ml
 +++ b/stdlib/hashtbl.ml
-@@ -284,7 +284,7 @@ module type SeededHashedType =
+@@ -292,7 +292,7 @@ module type SeededHashedType =
    sig
      type t
      val equal: t -> t -> bool
@@ -163,7 +163,7 @@
    end
  
  module type S =
-@@ -354,7 +354,7 @@ module MakeSeeded(H: SeededHashedType): (SeededS with type key = H.t) =
+@@ -362,7 +362,7 @@ module MakeSeeded(H: SeededHashedType): (SeededS with type key = H.t) =
      let copy = copy
  
      let key_index h key =
@@ -172,7 +172,7 @@
  
      let add h key data =
        let i = key_index h key in
-@@ -481,7 +481,7 @@ module Make(H: HashedType): (S with type key = H.t) =
+@@ -489,7 +489,7 @@ module Make(H: HashedType): (S with type key = H.t) =
      include MakeSeeded(struct
          type t = H.t
          let equal = H.equal
