--- a	2023-09-28 18:43:46.159873406 +0100
+++ b	2023-09-28 18:43:46.175874069 +0100
@@ -585,7 +585,7 @@
     trailing semicolon when it is printed vertically:
  
     {[
-@@ -1028,10 +1028,12 @@ val make_formatter :
+@@ -1029,10 +1029,12 @@ val make_formatter :
  (** [make_formatter out flush] returns a new formatter that outputs with
    function [out], and flushes with function [flush].
  
@@ -784,7 +784,7 @@
    *)
  
  external field : t -> int -> t = "%obj_field"
-@@ -132,40 +132,40 @@ module Ephemeron: sig
+@@ -134,40 +134,40 @@ module Ephemeron: sig
    (** return the number of keys *)
  
    val get_key: t -> int -> obj_t option
