--- a	2023-09-28 18:45:41.248471604 +0100
+++ b	2023-09-28 18:45:41.260472067 +0100
@@ -13,7 +13,7 @@
  ### Other libraries:
  
  - #11475: Make Unix terminal interface bindings domain-safe
-@@ -642,6 +637,11 @@ OCaml 5.0
+@@ -637,6 +632,11 @@ OCaml 5.0
  - #11516, #11524: Fix the `deprecated_mutable` attribute.
    (Chris Casinghino, review by Nicolás Ojeda Bär and Florian Angeletti)
  
@@ -22,6 +22,6 @@
 +  (Nicolás Ojeda Bär, Jeremy Yallop, report by Masayuki Takeda, review by Jeremy
 +  Yallop and Florian Angeletti)
 +
- OCaml 4.14.0 (28 March 2022)
- ----------------------------
+ OCaml 4.14 maintenance branch
+ -----------------------------
  
