--- a	2023-09-28 18:47:12.675911410 +0100
+++ b	2023-09-28 18:47:12.683911703 +0100
@@ -1,7 +1,7 @@
 diff --git a/lambda/matching.ml b/lambda/matching.ml
 --- a/lambda/matching.ml
 +++ b/lambda/matching.ml
-@@ -2888,7 +2888,7 @@ let combine_constructor loc arg pat_env cstr partial ctx def
+@@ -2936,7 +2936,7 @@ let combine_constructor loc arg pat_env cstr partial ctx def
                  (* This case is very frequent, it corresponds to
                     options and lists. *)
                  (* Keeping the Pisint test would make the bytecode
