--- a	2023-09-28 18:44:33.565807504 +0100
+++ b	2023-09-28 18:44:33.573807826 +0100
@@ -83,7 +83,7 @@
  AC_SUBST([flexdll_chain])
  AC_SUBST([PACKLD])
  AC_SUBST([stdlib_manpages])
-@@ -691,6 +694,10 @@ AS_IF([test x"$enable_shared" = "xno"],
+@@ -701,6 +704,10 @@ AS_IF([test x"$enable_shared" = "xno"],
      [AC_MSG_ERROR([Cannot build native Win32 with --disable-shared])])],
    [supports_shared_libraries=true])
  
@@ -94,7 +94,7 @@
  # Define flexlink chain and flags correctly for the different Windows ports
  AS_CASE([$host],
    [i686-*-cygwin],
-@@ -788,6 +795,11 @@ AS_CASE([$flexdir,$supports_shared_libraries,$flexlink,$host],
+@@ -798,6 +805,11 @@ AS_CASE([$flexdir,$supports_shared_libraries,$flexlink,$host],
    [,*,,*-w64-mingw32|,*,,*-pc-windows],
      [AC_MSG_ERROR([flexlink is required for native Win32])])
  
@@ -106,7 +106,7 @@
  AS_CASE([$cc_basename,$host],
    [*,x86_64-*-darwin*],
      [oc_ldflags='-Wl,-no_compact_unwind';
-@@ -798,7 +810,10 @@ AS_CASE([$cc_basename,$host],
+@@ -808,7 +820,10 @@ AS_CASE([$cc_basename,$host],
    [*,*-*-cygwin*],
      [common_cppflags="$common_cppflags -U_WIN32"
      AS_IF([$supports_shared_libraries],
@@ -118,7 +118,7 @@
        mkexe_cflags=''
        mkexe_ldflags_prefix='-link ']
        [mkexe_extra_flags=''
-@@ -810,7 +825,10 @@ AS_CASE([$cc_basename,$host],
+@@ -820,7 +835,10 @@ AS_CASE([$cc_basename,$host],
        [i686-*-*], [oc_dll_ldflags="-static-libgcc"])
      ostype="Win32"
      toolchain="mingw"
@@ -130,7 +130,7 @@
      mkexe_cflags=''
      mkexe_ldflags_prefix='-link '
      oc_ldflags='-municode'
-@@ -818,7 +836,10 @@ AS_CASE([$cc_basename,$host],
+@@ -828,7 +846,10 @@ AS_CASE([$cc_basename,$host],
    [*,*-pc-windows],
      [toolchain=msvc
      ostype="Win32"
@@ -142,7 +142,7 @@
      mkexe_cflags=''
      mkexe_ldflags_prefix='-link '
      mkexe_via_cc_ldflags_prefix='/link '
-@@ -830,9 +851,7 @@ AS_CASE([$cc_basename,$host],
+@@ -840,9 +861,7 @@ AS_CASE([$cc_basename,$host],
      [oc_ldflags='-brtl -bexpfull'
      AC_DEFINE([HAS_ARCH_CODE32], [1])],
    [gcc*,powerpc-*-linux*],
@@ -153,7 +153,7 @@
  
  ## Program to use to install files
  AC_PROG_INSTALL
-@@ -929,40 +948,44 @@ AS_IF([! $arch64],
+@@ -939,40 +958,44 @@ AS_IF([! $arch64],
  # Shared library support
  
  sharedlib_cflags=''
@@ -212,7 +212,7 @@
        rpath="-Wl,-rpath,"
        mksharedlibrpath="-Wl,-rpath,"
        supports_shared_libraries=true],
-@@ -971,19 +994,24 @@ AS_IF([test x"$enable_shared" != "xno"],
+@@ -981,19 +1004,24 @@ AS_IF([test x"$enable_shared" != "xno"],
        [sharedlib_cflags="-fPIC"
         AS_CASE([$cc_basename,$host],
             [*gcc*,powerpc-*-linux*],
@@ -242,7 +242,7 @@
  
  # Configure native dynlink
  
-@@ -2069,35 +2097,60 @@ unset ac_cv_header_flexdll_h
+@@ -2106,35 +2134,60 @@ unset ac_cv_header_flexdll_h
  # Just before config.status is generated, determine the final values for MKEXE,
  # MKDLL, MKMAINDLL and MKEXE_VIA_CC. The final variables controlling these are:
  # $mkexe - the linking commmand and munged CFLAGS + any extra flexlink flags
