--- a	2023-09-28 18:43:37.475512506 +0100
+++ b	2023-09-28 18:43:37.515514173 +0100
@@ -108,7 +108,7 @@
  AC_SUBST([flexdll_chain])
  AC_SUBST([flexlink_flags])
  AC_SUBST([PACKLD])
-@@ -372,37 +370,14 @@ AC_ARG_ENABLE([warn-error],
+@@ -381,37 +379,14 @@ AC_ARG_ENABLE([warn-error],
  AC_ARG_VAR([WINDOWS_UNICODE_MODE],
    [how to handle Unicode under Windows: ansi, compatible])
  
@@ -154,7 +154,7 @@
  
  AC_ARG_ENABLE([flat-float-array],
    [AS_HELP_STRING([--disable-flat-float-array],
-@@ -1947,15 +1922,6 @@ AS_IF([test x"$with_afl" = "xyes"],
+@@ -1981,15 +1956,6 @@ AS_IF([test x"$with_afl" = "xyes"],
    [afl=true],
    [afl=false])
  
@@ -439,7 +439,7 @@
       in
       match Env.open_pers_signature modname env with
       | Error `Not_found ->
-@@ -3442,7 +3442,7 @@ let failure_handler ~scopes loc ~failer () =
+@@ -3490,7 +3490,7 @@ let failure_handler ~scopes loc ~failer () =
      let sloc = Scoped_location.of_location ~scopes loc in
      let slot =
        transl_extension_path sloc
