--- a	2023-09-28 18:47:01.219489461 +0100
+++ b	2023-09-28 18:47:01.239490200 +0100
@@ -9,9 +9,9 @@
 +  types.
 +  (Chris Casinghino, review by Gabriel Scherer)
 +
- OCaml 5.0.0 (15 December 2022)
- ------------------------------
- 
+ - #12264, #12289: Fix compact_allocate to avoid a pathological case
+   that causes very slow compaction.
+   (Damien Doligez, report by Arseniy Alekseyev, review by Sadiq Jaffer)
 diff --git a/lambda/translcore.ml b/lambda/translcore.ml
 --- a/lambda/translcore.ml
 +++ b/lambda/translcore.ml
