--- a	2023-09-28 18:47:14.431975870 +0100
+++ b	2023-09-28 18:47:14.447976458 +0100
@@ -48,7 +48,7 @@
 diff --git a/bytecomp/dll.ml b/bytecomp/dll.ml
 --- a/bytecomp/dll.ml
 +++ b/bytecomp/dll.ml
-@@ -164,9 +164,6 @@ let ld_library_path_contents () =
+@@ -172,9 +172,6 @@ let ld_library_path_contents () =
    | s ->
        Misc.split_path_contents s
  
@@ -58,7 +58,7 @@
  (* Initialization for separate compilation *)
  
  let init_compile nostdlib =
-@@ -176,10 +173,10 @@ let init_compile nostdlib =
+@@ -184,10 +181,10 @@ let init_compile nostdlib =
  
  (* Initialization for linking in core (dynlink or toplevel) *)
  
@@ -128,7 +128,7 @@
 diff --git a/tools/dumpobj.ml b/tools/dumpobj.ml
 --- a/tools/dumpobj.ml
 +++ b/tools/dumpobj.ml
-@@ -477,18 +477,13 @@ let dump_obj ic =
+@@ -479,18 +479,13 @@ let dump_obj ic =
    seek_in ic cu.cu_pos;
    print_code ic cu.cu_codesize
  
