--- a	2023-09-28 18:44:56.302714648 +0100
+++ b	2023-09-28 18:44:56.322715441 +0100
@@ -109,15 +109,15 @@
  AC_SUBST([cmxs])
  AC_SUBST([debug_runtime])
  AC_SUBST([instrumented_runtime])
-@@ -180,6 +181,7 @@ AC_SUBST([QS])
+@@ -182,6 +183,7 @@ AC_SUBST([QS])
  AC_CONFIG_FILES([Makefile.build_config])
  AC_CONFIG_FILES([Makefile.config])
  AC_CONFIG_FILES([stdlib/sys.ml])
 +AC_CONFIG_FILES([otherlibs/dynlink/META])
  AC_CONFIG_FILES([manual/src/version.tex])
  AC_CONFIG_FILES([manual/src/html_processing/src/common.ml])
- AC_CONFIG_HEADERS([runtime/caml/m.h])
-@@ -1204,6 +1206,9 @@ AS_IF([$native_compiler],
+ AC_CONFIG_FILES([tools/eventlog_metadata])
+@@ -1216,6 +1218,9 @@ AS_IF([$native_compiler],
  AS_IF([! $native_compiler], [natdynlink=false])
  
  AS_IF([$natdynlink], [cmxs="cmxs"], [cmxs="cmx"])
