--- a	2023-09-28 18:43:54.736227762 +0100
+++ b	2023-09-28 18:43:54.748228257 +0100
@@ -1,7 +1,7 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -635,6 +635,11 @@ OCaml 4.14.0
+@@ -644,6 +644,11 @@ OCaml 4.14.0
    Gentoo and Alpine).
    (Alex Fan, review by Gabriel Scherer)
  
@@ -16,7 +16,7 @@
 diff --git a/asmcomp/amd64/emit.mlp b/asmcomp/amd64/emit.mlp
 --- a/asmcomp/amd64/emit.mlp
 +++ b/asmcomp/amd64/emit.mlp
-@@ -826,7 +826,10 @@ let emit_instr env fallthrough i =
+@@ -851,7 +851,10 @@ let emit_instr env fallthrough i =
        done;
        emit_named_text_section env.f.fun_name
    | Lentertrap ->
