--- a	2023-09-28 18:45:16.063492880 +0100
+++ b	2023-09-28 18:45:16.079493506 +0100
@@ -14,7 +14,7 @@
 diff --git a/stdlib/hashtbl.ml b/stdlib/hashtbl.ml
 --- a/stdlib/hashtbl.ml
 +++ b/stdlib/hashtbl.ml
-@@ -421,7 +421,7 @@ module MakeSeeded(H: SeededHashedType): (SeededS with type key = H.t) =
+@@ -429,7 +429,7 @@ module MakeSeeded(H: SeededHashedType): (SeededS with type key = H.t) =
                    if H.equal key k3 then Some d3 else find_rec_opt key next3
  
      let find_all h key =
@@ -23,7 +23,7 @@
        | Empty ->
            []
        | Cons{key=k; data=d; next} ->
-@@ -571,7 +571,7 @@ let find_opt h key =
+@@ -579,7 +579,7 @@ let find_opt h key =
                if compare key k3 = 0 then Some d3 else find_rec_opt key next3
  
  let find_all h key =
