--- a	2023-09-28 18:43:48.103953910 +0100
+++ b	2023-09-28 18:43:48.115954406 +0100
@@ -1,7 +1,7 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -124,6 +124,9 @@ Working version
+@@ -127,6 +127,9 @@ Working version
  
  ### Bug fixes:
  
@@ -16,7 +16,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -1172,16 +1172,16 @@ AS_IF([test -z "$PARTIALLD"],
+@@ -1184,16 +1184,16 @@ AS_IF([test -z "$PARTIALLD"],
  # Disable PIE at link time when ocamlopt does not produce position-independent
  # code and the system produces PIE executables by default and demands PIC
  # object files to do so.
