--- a	2023-09-28 18:48:15.206172988 +0100
+++ b	2023-09-28 18:48:15.222173558 +0100
@@ -60,9 +60,9 @@
 diff --git a/otherlibs/systhreads/thread.ml b/otherlibs/systhreads/thread.ml
 --- a/otherlibs/systhreads/thread.ml
 +++ b/otherlibs/systhreads/thread.ml
-@@ -71,29 +71,12 @@ let create fn arg =
- let exit () =
-   raise Exit
+@@ -88,32 +88,13 @@ let exit () =
+ let kill th = invalid_arg "Thread.kill: not implemented"
+ (* BACKPORT END *)
  
 -(* Preemption *)
 -
@@ -76,8 +76,11 @@
 -  | _       -> Sys.sigvtalrm
 -
  let () =
+   Sys.set_signal preempt_signal (Sys.Signal_handle preempt);
    thread_initialize ();
+-(* BACKPORT
 -  Sys.set_signal preempt_signal (Sys.Signal_handle preempt);
+-*)
 -  (* Callback in [caml_shutdown], when the last domain exits. *)
 -  Callback.register "Thread.at_shutdown" (fun () ->
 -    thread_cleanup();
@@ -95,15 +98,15 @@
 diff --git a/otherlibs/unix/fork.c b/otherlibs/unix/fork.c
 --- a/otherlibs/unix/fork.c
 +++ b/otherlibs/unix/fork.c
-@@ -38,7 +38,7 @@ CAMLprim value caml_unix_fork(value unit)
-   int ret;
+@@ -45,7 +45,7 @@ CAMLprim value caml_unix_fork(value unit)
+ #if 0 /* BACKPORT */
    if (caml_domain_is_multicore()) {
      caml_failwith
 -      ("Unix.fork may not be called while other domains were created");
 +      ("Unix.fork may not be called after any domain has been spawned");
    }
+ #endif
  
-   ret = fork();
 diff --git a/otherlibs/unix/unix.mli b/otherlibs/unix/unix.mli
 --- a/otherlibs/unix/unix.mli
 +++ b/otherlibs/unix/unix.mli
@@ -151,7 +154,7 @@
 diff --git a/testsuite/tests/lib-unix/common/multicore_fork_domain_alone.ml b/testsuite/tests/lib-unix/common/multicore_fork_domain_alone.ml
 --- a/testsuite/tests/lib-unix/common/multicore_fork_domain_alone.ml
 +++ b/testsuite/tests/lib-unix/common/multicore_fork_domain_alone.ml
-@@ -12,16 +12,11 @@
+@@ -14,16 +14,11 @@
  (* on Multicore, fork is not allowed is another domain is, and was running. *)
  (* this test checks that we can't fork if a domain is currently running. *)
  
@@ -183,7 +186,7 @@
 diff --git a/testsuite/tests/lib-unix/common/multicore_fork_domain_alone2.ml b/testsuite/tests/lib-unix/common/multicore_fork_domain_alone2.ml
 --- a/testsuite/tests/lib-unix/common/multicore_fork_domain_alone2.ml
 +++ b/testsuite/tests/lib-unix/common/multicore_fork_domain_alone2.ml
-@@ -12,15 +12,11 @@
+@@ -14,15 +14,11 @@
  (* on Multicore, fork is not allowed is another domain is, and was running. *)
  (* this test checks that we can't fork if another domain ran before. *)
  
