--- a	2023-09-28 18:44:41.614130076 +0100
+++ b	2023-09-28 18:44:41.622130396 +0100
@@ -1,7 +1,7 @@
 diff --git a/testsuite/tests/parallel/domain_serial_spawn_burn.ml b/testsuite/tests/parallel/domain_serial_spawn_burn.ml
 --- a/testsuite/tests/parallel/domain_serial_spawn_burn.ml
 +++ b/testsuite/tests/parallel/domain_serial_spawn_burn.ml
-@@ -11,12 +11,17 @@ let test_size =
+@@ -13,12 +13,17 @@ let test_size =
    try int_of_string (Sys.getenv "OCAML_TEST_SIZE")
    with Not_found | Failure _ -> 0
  
