--- a	2023-09-28 18:43:19.466757288 +0100
+++ b	2023-09-28 18:43:19.494758470 +0100
@@ -11,6 +11,16 @@
  ### Compiler user-interface and warnings:
  
  ### Internal/compiler-libs changes:
+@@ -217,6 +220,9 @@ OCaml 4.14.0
+ - #10281, #10685: Add description of C compiler on macOS and Windows platforms.
+   (Dong An, review by Xavier Leroy and David Allsopp)
+ 
++- #10589: Fix many typos (excess/inconsistent spaces) in the HTML manual.
++  (Wiktor Kuchta, review by Florian Angeletti)
++
+ - #10605: manual, name few css classes to ease styling and maintainability.
+   (Florian Angeletti, review by Wiktor Kuchta and Gabriel Scherer)
+ 
 diff --git a/manual/src/cmds/unified-options.etex b/manual/src/cmds/unified-options.etex
 --- a/manual/src/cmds/unified-options.etex
 +++ b/manual/src/cmds/unified-options.etex
