--- a	2023-09-28 18:44:22.133346479 +0100
+++ b	2023-09-28 18:44:22.157347450 +0100
@@ -77,7 +77,7 @@
  AC_SUBST([mklib])
  AC_SUBST([supports_shared_libraries])
  AC_SUBST([natdynlink])
-@@ -472,32 +470,24 @@ AS_CASE([$enable_dependency_generation],
+@@ -482,32 +480,24 @@ AS_CASE([$enable_dependency_generation],
        [compute_deps=true])],
      [compute_deps=false])])
  
@@ -256,7 +256,7 @@
  let default_rpath = "%%RPATH%%"
  let mksharedlibrpath = "%%MKSHAREDLIBRPATH%%"
  let ar = "%%ARCMD%%"
-@@ -178,7 +177,6 @@ let configuration_variables =
+@@ -183,7 +182,6 @@ let configuration_variables =
    p "bytecomp_c_libraries" bytecomp_c_libraries;
    p "native_c_libraries" native_c_libraries;
    p "native_pack_linker" native_pack_linker;
