--- a	2023-09-28 18:45:32.816145510 +0100
+++ b	2023-09-28 18:45:32.828145976 +0100
@@ -55,7 +55,7 @@
  AC_SUBST([long_shebang])
  AC_SUBST([shebangscripts])
  AC_SUBST([AR])
-@@ -779,29 +782,30 @@ AS_IF([test x"$enable_shared" = "xno"],
+@@ -791,29 +794,30 @@ AS_IF([test x"$enable_shared" = "xno"],
  # The flexlink command
  
  flexlink_cmd=flexlink
@@ -92,7 +92,7 @@
    [flexdll_chain=''])
  
  AS_IF([test x"$supports_shared_libraries" != 'xfalse'], [
-@@ -893,12 +897,13 @@ AS_CASE([$cc_basename,$host],
+@@ -905,12 +909,13 @@ AS_CASE([$cc_basename,$host],
    [*,*-*-cygwin*],
      [common_cppflags="$common_cppflags -U_WIN32"
      AS_IF([$supports_shared_libraries],
@@ -109,7 +109,7 @@
        [mkexe_extra_flags=''
        oc_ldflags='-Wl,--stack,16777216']
      )
-@@ -908,25 +913,29 @@ AS_CASE([$cc_basename,$host],
+@@ -920,25 +925,29 @@ AS_CASE([$cc_basename,$host],
        [i686-*-*], [oc_dll_ldflags="-static-libgcc"])
      ostype="Win32"
      toolchain="mingw"
@@ -145,7 +145,7 @@
      mkexedebugflag=''],
    [*,x86_64-*-linux*],
      AC_DEFINE([HAS_ARCH_CODE32], [1]),
-@@ -1060,20 +1069,19 @@ AS_IF([test x"$enable_shared" != "xno"],
+@@ -1072,20 +1081,19 @@ AS_IF([test x"$enable_shared" != "xno"],
        [mkdll_flags='-shared -undefined dynamic_lookup'
        supports_shared_libraries=true],
      [*-*-mingw32],
@@ -175,7 +175,7 @@
      [powerpc-ibm-aix*],
        [AS_CASE([$ocaml_cv_cc_vendor],
                 [xlc*],
-@@ -2243,7 +2251,7 @@ AC_CONFIG_COMMANDS_PRE([
+@@ -2293,7 +2301,7 @@ AC_CONFIG_COMMANDS_PRE([
  # $mkdll - the full linking command for DLLs
  # $mkdll_exp: same as above but expanded
  # $mkmaindll - the full linking command for main program in DLL
