--- a	2023-09-28 18:44:27.909579830 +0100
+++ b	2023-09-28 18:44:27.917580153 +0100
@@ -1,10 +1,13 @@
 diff --git a/testsuite/tests/weak-ephe-final/weaklifetime_par.ml b/testsuite/tests/weak-ephe-final/weaklifetime_par.ml
 --- a/testsuite/tests/weak-ephe-final/weaklifetime_par.ml
 +++ b/testsuite/tests/weak-ephe-final/weaklifetime_par.ml
-@@ -1,4 +1,5 @@
--(* TEST
+@@ -2,6 +2,9 @@
+    * skip
+    reason = "OCaml 5 only"
+ *)
 +(* This test is currently disabled, 
 +   until the random failures have been investigated.
- *)
++*)
  
  let () = Random.self_init ()
+ 
