--- a	2023-09-28 18:48:35.774902508 +0100
+++ b	2023-09-28 18:48:35.782902791 +0100
@@ -1,7 +1,7 @@
 diff --git a/lambda/matching.ml b/lambda/matching.ml
 --- a/lambda/matching.ml
 +++ b/lambda/matching.ml
-@@ -3736,8 +3736,14 @@ let for_let ~scopes loc param pat body =
+@@ -3784,8 +3784,14 @@ let for_let ~scopes loc param pat body =
        (* This eliminates a useless variable (and stack slot in bytecode)
           for "let _ = ...". See #6865. *)
        Lsequence (param, body)
