--- a	2023-09-28 18:47:08.299750522 +0100
+++ b	2023-09-28 18:47:08.323751405 +0100
@@ -25,8 +25,6 @@
  ### Manual and documentation:
  
  - #9430, #11291: Document the general desugaring rules for binding operators.
-diff --git a/boot/ocamlc b/boot/ocamlc
-Binary files a/boot/ocamlc and b/boot/ocamlc differ
 diff --git a/bytecomp/bytepackager.ml b/bytecomp/bytepackager.ml
 --- a/bytecomp/bytepackager.ml
 +++ b/bytecomp/bytepackager.ml
@@ -45,7 +43,7 @@
 diff --git a/bytecomp/emitcode.ml b/bytecomp/emitcode.ml
 --- a/bytecomp/emitcode.ml
 +++ b/bytecomp/emitcode.ml
-@@ -416,8 +416,9 @@ let to_file outchan unit_name objfile ~required_globals code =
+@@ -420,8 +420,9 @@ let to_file outchan unit_name objfile ~required_globals code =
          (Filename.dirname (Location.absolute_path objfile))
          !debug_dirs;
        let p = pos_out outchan in
@@ -62,7 +60,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -471,6 +471,10 @@ AC_ARG_WITH([flexdll],
+@@ -482,6 +482,10 @@ AC_ARG_WITH([flexdll],
      [bootstrap FlexDLL from the given sources])],
    [AS_IF([test x"$withval" = 'xyes'],[with_flexdll=flexdll])])
  
@@ -73,7 +71,7 @@
  AS_IF([test x"$enable_unix_lib" = "xno"],
    [AS_IF([test x"$enable_debugger" = "xyes"],
      [AC_MSG_ERROR([replay debugger requires the unix library])],
-@@ -1929,6 +1933,42 @@ AC_COMPILE_IFELSE(
+@@ -1943,6 +1947,42 @@ AC_COMPILE_IFELSE(
          pthread_getaffinity_np(pthread_self(), sizeof(cs), &cs);]])],
      [AC_DEFINE([HAS_BSD_GETAFFINITY_NP])])])
  
@@ -143,7 +141,7 @@
 diff --git a/stdlib/.depend b/stdlib/.depend
 --- a/stdlib/.depend
 +++ b/stdlib/.depend
-@@ -488,9 +488,11 @@ stdlib__Map.cmx : map.ml \
+@@ -444,9 +444,11 @@ stdlib__Map.cmx : map.ml \
  stdlib__Map.cmi : map.mli \
      stdlib__Seq.cmi
  stdlib__Marshal.cmo : marshal.ml \
@@ -507,7 +505,7 @@
 diff --git a/utils/config.common.ml b/utils/config.common.ml
 --- a/utils/config.common.ml
 +++ b/utils/config.common.ml
-@@ -74,7 +74,7 @@ type configuration_value =
+@@ -79,7 +79,7 @@ type configuration_value =
    | Int of int
    | Bool of bool
  
@@ -516,7 +514,7 @@
    let p x v = (x, String v) in
    let p_int x v = (x, Int v) in
    let p_bool x v = (x, Bool v) in
-@@ -122,6 +122,7 @@ let configuration_variables =
+@@ -127,6 +127,7 @@ let configuration_variables =
    p_bool "supports_shared_libraries" supports_shared_libraries;
    p_bool "native_dynlink" native_dynlink;
    p_bool "naked_pointers" naked_pointers;
@@ -524,7 +522,7 @@
  
    p "exec_magic_number" exec_magic_number;
    p "cmi_magic_number" cmi_magic_number;
-@@ -147,11 +148,11 @@ let print_config_value oc = function
+@@ -152,11 +153,11 @@ let print_config_value oc = function
  let print_config oc =
    let print (x, v) =
      Printf.fprintf oc "%s: %a\n" x print_config_value v in
