--- a	2023-09-28 18:48:07.009880371 +0100
+++ b	2023-09-28 18:48:07.021880800 +0100
@@ -22,18 +22,17 @@
  - #12037: get_extern_state potential NULL dereference.
    (Alexander Skvortsov, report by Török Edwin,
     design by Gabriel Scherer, Xavier Leroy)
-@@ -188,10 +187,6 @@ Working version
+@@ -188,9 +187,6 @@ Working version
    (Vincent Laviron, report by François Pottier, review by Nathanaëlle Courant
    and Gabriel Scherer)
  
 -- #12037: Fix some data races by using volatile when necessary
 -  (Fabrice Buoro and Olivier Nicole, review by Guillaume Munch-Maccagnoni,
 -  Gabriel Scherer and Luc Maranget)
--
+ 
  OCaml 5.1.0
  ---------------
- 
-@@ -1006,13 +1001,14 @@ Some of those changes will benefit all OCaml packages.
+@@ -1006,13 +1002,14 @@ Some of those changes will benefit all OCaml packages.
  - #11879: Bugfix for Ctype.nondep_type
    (Stephen Dolan, review by Gabriel Scherer)
  
@@ -51,7 +50,7 @@
  - #12046: Flush stderr when tracing the parser
    (Hugo Heuzard, review by David Allsopp and Nicolás Ojeda Bär)
  
-@@ -1060,6 +1056,10 @@ Some of those changes will benefit all OCaml packages.
+@@ -1060,6 +1057,10 @@ Some of those changes will benefit all OCaml packages.
    emitting stack reallocation code before the body of the function.
    (Xavier Leroy, review by KC Sivaramakrishnan)
  
