--- a	2023-09-28 18:48:26.006556898 +0100
+++ b	2023-09-28 18:48:26.022557466 +0100
@@ -24,7 +24,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -1235,13 +1235,19 @@ arch=none
+@@ -1248,13 +1248,19 @@ arch=none
  model=default
  system=unknown
  
@@ -47,15 +47,18 @@
    [[i[3456]86-*-haiku*]],
      [arch=i386; system=beos],
    [[i[3456]86-*-cygwin]],
-@@ -1255,24 +1261,24 @@ AS_CASE([$host],
+@@ -1268,26 +1274,26 @@ AS_CASE([$host],
    [x86_64-pc-windows],
-     [arch=amd64; system=win64],
+     [has_native_backend=yes; arch=amd64; system=win64],
    [[powerpc64le*-*-linux*]],
--    [has_native_backend=yes; arch=power; model=ppc64le; system=elf],
-+    [has_native_backend=yes; arch=power; model=ppc64le; system=linux],
+-    [arch=power; model=ppc64le; system=elf],
++    [arch=power; model=ppc64le; system=linux],
+   [[powerpc*-*-linux*]],
+-    [arch=power; model=ppc; system=elf],
++    [arch=power; model=ppc; system=linux],
    [[s390x*-*-linux*]],
--    [has_native_backend=yes; arch=s390x; model=z10; system=elf],
-+    [has_native_backend=yes; arch=s390x; model=z10; system=linux],
+-    [arch=s390x; model=z10; system=elf],
++    [arch=s390x; model=z10; system=linux],
    # expected to match "gnueabihf" as well as "musleabihf"
    [armv6*-*-linux-*eabihf],
 -    [arch=arm; model=armv6; system=linux_eabihf],
@@ -81,7 +84,7 @@
    [armv6*-*-freebsd*],
      [arch=arm; model=armv6; system=freebsd],
    [earmv6*-*-netbsd*],
-@@ -1280,17 +1286,17 @@ AS_CASE([$host],
+@@ -1295,17 +1301,17 @@ AS_CASE([$host],
    [earmv7*-*-netbsd*],
      [arch=arm; model=armv7; system=netbsd],
    [armv5te*-*-linux-*eabi],
