--- a	2023-09-28 18:44:33.281796092 +0100
+++ b	2023-09-28 18:44:33.289796414 +0100
@@ -34,15 +34,15 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -169,6 +169,7 @@ AC_SUBST([PACKLD])
- AC_SUBST([stdlib_manpages])
- AC_SUBST([compute_deps])
+@@ -171,6 +171,7 @@ AC_SUBST([compute_deps])
+ AC_SUBST([naked_pointers])
+ AC_SUBST([naked_pointers_checker])
  AC_SUBST([force_instrumented_runtime])
 +AC_SUBST([QS])
  
  ## Generated files
  
-@@ -192,6 +193,9 @@ m4_if([OCAML__VERSION_EXTRA],[], [],
+@@ -194,6 +195,9 @@ m4_if([OCAML__VERSION_EXTRA],[], [],
  AC_DEFINE([OCAML_VERSION], [OCAML__VERSION_NUMBER])
  AC_DEFINE([OCAML_VERSION_STRING], ["][OCAML__VERSION]["])
  
