--- a	2023-09-28 18:48:27.614613895 +0100
+++ b	2023-09-28 18:48:27.618614037 +0100
@@ -1,9 +1,9 @@
 diff --git a/testsuite/tests/lib-runtime-events/test_caml.ml b/testsuite/tests/lib-runtime-events/test_caml.ml
 --- a/testsuite/tests/lib-runtime-events/test_caml.ml
 +++ b/testsuite/tests/lib-runtime-events/test_caml.ml
-@@ -1,14 +1,17 @@
- (* TEST
-  include runtime_events;
+@@ -3,14 +3,17 @@
+  reason = "OCaml 5 only";
+  skip;
  *)
 +
 +(* Tests that:
@@ -25,7 +25,7 @@
  
  let got_start = ref false
  
-@@ -16,24 +19,32 @@ let lifecycle domain_id ts lifecycle_event data =
+@@ -18,24 +21,32 @@ let lifecycle domain_id ts lifecycle_event data =
      match lifecycle_event with
      | EV_RING_START ->
          begin
@@ -65,7 +65,7 @@
          end
      | _ -> ()
  
-@@ -41,22 +52,30 @@ let runtime_end domain_id ts phase =
+@@ -43,22 +54,30 @@ let runtime_end domain_id ts phase =
      match phase with
      | EV_MAJOR_FINISH_CYCLE ->
          begin
@@ -100,7 +100,7 @@
  
  let () =
      let list_ref = ref [] in
-@@ -66,14 +85,14 @@ let () =
+@@ -68,14 +87,14 @@ let () =
                                      ~lost_events ()
      in
      for epoch = 1 to epochs do
