--- a	2023-09-28 18:45:25.907877164 +0100
+++ b	2023-09-28 18:45:25.927877942 +0100
@@ -14,7 +14,7 @@
 diff --git a/stdlib/hashtbl.ml b/stdlib/hashtbl.ml
 --- a/stdlib/hashtbl.ml
 +++ b/stdlib/hashtbl.ml
-@@ -447,13 +447,14 @@ module MakeSeeded(H: SeededHashedType): (SeededS with type key = H.t) =
+@@ -455,13 +455,14 @@ module MakeSeeded(H: SeededHashedType): (SeededS with type key = H.t) =
          if h.size > Array.length h.data lsl 1 then resize key_index h
        end
  
@@ -33,7 +33,7 @@
  
      let add_seq tbl i =
        Seq.iter (fun (k,v) -> add tbl k v) i
-@@ -597,13 +598,14 @@ let replace h key data =
+@@ -605,13 +606,14 @@ let replace h key data =
      if h.size > Array.length h.data lsl 1 then resize key_index h
    end
  
@@ -59,17 +59,17 @@
  Raised by primitive operation at Backtrace2.run in file "backtrace2.ml", line 62, characters 14-22
  test_Not_found
  Uncaught exception Not_found
--Raised at Stdlib__Hashtbl.find in file "hashtbl.ml", line 541, characters 13-28
-+Raised at Stdlib__Hashtbl.find in file "hashtbl.ml", line 542, characters 13-28
+-Raised at Stdlib__Hashtbl.find in file "hashtbl.ml", line 549, characters 13-28
++Raised at Stdlib__Hashtbl.find in file "hashtbl.ml", line 550, characters 13-28
  Called from Backtrace2.test_Not_found in file "backtrace2.ml", line 43, characters 9-42
  Re-raised at Backtrace2.test_Not_found in file "backtrace2.ml", line 43, characters 61-70
  Called from Backtrace2.run in file "backtrace2.ml", line 62, characters 11-23
-@@ -50,7 +50,7 @@ Called from CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line
- Re-raised at CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 56, characters 4-11
+@@ -50,7 +50,7 @@ Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", lin
+ Re-raised at CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 124, characters 4-11
  Called from Backtrace2.run in file "backtrace2.ml", line 62, characters 11-23
  Uncaught exception Not_found
--Raised at Stdlib__Hashtbl.find in file "hashtbl.ml", line 541, characters 13-28
-+Raised at Stdlib__Hashtbl.find in file "hashtbl.ml", line 542, characters 13-28
+-Raised at Stdlib__Hashtbl.find in file "hashtbl.ml", line 549, characters 13-28
++Raised at Stdlib__Hashtbl.find in file "hashtbl.ml", line 550, characters 13-28
  Called from Backtrace2.test_lazy.exception_raised_internally in file "backtrace2.ml", line 50, characters 8-41
- Re-raised at CamlinternalLazy.do_force_block.(fun) in file "camlinternalLazy.ml", line 54, characters 43-50
- Called from CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 49, characters 17-27
+ Re-raised at CamlinternalLazy.force_lazy_block.(fun) in file "camlinternalLazy.ml", line 123, characters 56-63
+ Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 119, characters 17-27
