--- a	2023-09-28 18:47:40.560928384 +0100
+++ b	2023-09-28 18:47:40.580929109 +0100
@@ -9,14 +9,13 @@
  OCaml 5.1.0
  ---------------
  
-@@ -839,6 +838,10 @@ OCaml 5.1.0
+@@ -839,6 +838,9 @@ OCaml 5.1.0
  - #12178: Fix runtime events consumer poll function returning an invalid value
    instead of an OCaml integer value. (Lucas Pluvinage)
  
 +- #12252: Fix shared library build error on RISC-V.
 +  (Edwin Török, review by Nicolás Ojeda Bär and Xavier Leroy)
 +
-+
  OCaml 5.0.0 (15 December 2022)
  ------------------------------
  
