--- a	2023-09-28 18:43:40.451636423 +0100
+++ b	2023-09-28 18:43:40.471637255 +0100
@@ -17,14 +17,16 @@
 diff --git a/otherlibs/systhreads/thread.ml b/otherlibs/systhreads/thread.ml
 --- a/otherlibs/systhreads/thread.ml
 +++ b/otherlibs/systhreads/thread.ml
-@@ -85,9 +85,9 @@ let preempt_signal =
+@@ -100,11 +100,11 @@ let preempt_signal =
    | _       -> Sys.sigvtalrm
  
  let () =
 -  Sys.set_signal preempt_signal (Sys.Signal_handle preempt);
--  Domain.at_startup thread_initialize_domain;
-   thread_initialize ();
-+  Domain.at_startup thread_initialize_domain;
++  thread_initialize ();
+ (* BACKPORT
+   Domain.at_startup thread_initialize_domain;
+ *)
+-  thread_initialize ();
 +  Sys.set_signal preempt_signal (Sys.Signal_handle preempt);
    Callback.register "Thread.at_shutdown" (fun () ->
      thread_cleanup();
