--- a	2023-09-28 18:43:42.963740830 +0100
+++ b	2023-09-28 18:43:42.967740996 +0100
@@ -12,8 +12,8 @@
 diff --git a/otherlibs/unix/signals.c b/otherlibs/unix/signals.c
 --- a/otherlibs/unix/signals.c
 +++ b/otherlibs/unix/signals.c
-@@ -68,7 +68,7 @@ CAMLprim value unix_sigprocmask(value vaction, value vset)
-   retcode = sigprocmask(how, &set, &oldset);
+@@ -85,7 +85,7 @@ CAMLprim value unix_sigprocmask(value vaction, value vset)
+ /* BACKPORT END */
    caml_leave_blocking_section();
    /* Run any handlers for just-unmasked pending signals */
 -  caml_process_pending_signals();
