--- a	2023-09-28 18:45:46.692681303 +0100
+++ b	2023-09-28 18:45:46.704681765 +0100
@@ -11,13 +11,13 @@
  OCaml 5.0
  ---------
  
-@@ -659,6 +656,9 @@ OCaml 5.0
+@@ -654,6 +651,9 @@ OCaml 5.0
    (Nicolás Ojeda Bär, Jeremy Yallop, report by Masayuki Takeda, review by Jeremy
    Yallop and Florian Angeletti)
  
 +- #11587: Prevent integer comparison from being used on pointers
 +  (Vincent Laviron, review by Gabriel Scherer)
 +
- OCaml 4.14.0 (28 March 2022)
- ----------------------------
+ OCaml 4.14 maintenance branch
+ -----------------------------
  
