--- a	2023-09-28 18:48:02.473717942 +0100
+++ b	2023-09-28 18:48:02.493718659 +0100
@@ -16,7 +16,7 @@
 diff --git a/otherlibs/unix/unixsupport.h b/otherlibs/unix/unixsupport.h
 --- a/otherlibs/unix/unixsupport.h
 +++ b/otherlibs/unix/unixsupport.h
-@@ -100,13 +100,10 @@ extern void caml_win32_maperr(DWORD errcode);
+@@ -109,13 +109,10 @@ extern void caml_win32_maperr(DWORD errcode);
  extern value caml_unix_error_of_code (int errcode);
  extern int caml_unix_code_of_unix_error (value error);
  
