--- a	2023-09-28 18:46:52.279158457 +0100
+++ b	2023-09-28 18:46:52.299159199 +0100
@@ -70,6 +70,6 @@
 +- #11879: Bugfix for Ctype.nondep_type
 +  (Stephen Dolan, review by Gabriel Scherer)
 +
- OCaml 5.0
- ---------
- 
+ - #12264, #12289: Fix compact_allocate to avoid a pathological case
+   that causes very slow compaction.
+   (Damien Doligez, report by Arseniy Alekseyev, review by Sadiq Jaffer)
