--- a	2023-09-28 18:45:18.307580675 +0100
+++ b	2023-09-28 18:45:18.331581613 +0100
@@ -12,10 +12,6 @@
  - #11364: Allow `make -C testsuite promote` to take `TEST` and `LIST` variables
    (Antal Spector-Zabusky, review by Gabriel Scherer and David Allsopp)
  
-diff --git a/boot/ocamlc b/boot/ocamlc
-Binary files a/boot/ocamlc and b/boot/ocamlc differ
-diff --git a/boot/ocamllex b/boot/ocamllex
-Binary files a/boot/ocamllex and b/boot/ocamllex differ
 diff --git a/testsuite/tests/typing-misc/variance.ml b/testsuite/tests/typing-misc/variance.ml
 --- a/testsuite/tests/typing-misc/variance.ml
 +++ b/testsuite/tests/typing-misc/variance.ml
