--- a	2023-09-28 18:48:04.781800631 +0100
+++ b	2023-09-28 18:48:04.797801204 +0100
@@ -1,13 +1,14 @@
 diff --git a/Changes b/Changes
 --- a/Changes
 +++ b/Changes
-@@ -178,6 +178,9 @@ Working version
+@@ -178,6 +178,10 @@ Working version
    (Vincent Laviron, report by François Pottier, review by Nathanaëlle Courant
    and Gabriel Scherer)
  
 +- #12037: Fix some data races by using volatile when necessary
 +  (Fabrice Buoro and Olivier Nicole, review by Guillaume Munch-Maccagnoni,
 +  Gabriel Scherer and Luc Maranget)
- 
++
  OCaml 5.1.0
  ---------------
+ 
