--- a	2023-09-28 18:48:23.138455132 +0100
+++ b	2023-09-28 18:48:23.158455842 +0100
@@ -14,7 +14,7 @@
 diff --git a/stdlib/gc.ml b/stdlib/gc.ml
 --- a/stdlib/gc.ml
 +++ b/stdlib/gc.ml
-@@ -123,6 +123,7 @@ let delete_alarm a = Atomic.set a false
+@@ -129,6 +129,7 @@ let delete_alarm a = Atomic.set a false
  
  module Memprof =
    struct
@@ -22,7 +22,7 @@
      type allocation_source = Normal | Marshal | Custom
      type allocation =
        { n_samples : int;
-@@ -147,7 +148,7 @@ module Memprof =
+@@ -153,7 +154,7 @@ module Memprof =
      }
  
      external c_start :
@@ -31,7 +31,7 @@
        = "caml_memprof_start"
  
      let start
-@@ -157,4 +158,6 @@ module Memprof =
+@@ -163,4 +164,6 @@ module Memprof =
        c_start sampling_rate callstack_size tracker
  
      external stop : unit -> unit = "caml_memprof_stop"
