--- a	2023-09-28 18:45:06.043099409 +0100
+++ b	2023-09-28 18:45:06.059100039 +0100
@@ -19,7 +19,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -870,6 +870,7 @@ AS_IF([test "x$ac_cv_lib_m_cos" = xyes ], [mathlib="-lm"], [mathlib=""])
+@@ -882,6 +882,7 @@ AS_IF([test "x$ac_cv_lib_m_cos" = xyes ], [mathlib="-lm"], [mathlib=""])
  AC_CHECK_HEADER([math.h])
  AC_CHECK_HEADERS([unistd.h],[AC_DEFINE([HAS_UNISTD])])
  AC_CHECK_HEADER([stdint.h],[AC_DEFINE([HAS_STDINT_H])])
