--- a	2023-09-28 18:47:37.636822394 +0100
+++ b	2023-09-28 18:47:37.652822975 +0100
@@ -250,7 +250,7 @@
 diff --git a/otherlibs/unix/unixsupport_win32.c b/otherlibs/unix/unixsupport_win32.c
 --- a/otherlibs/unix/unixsupport_win32.c
 +++ b/otherlibs/unix/unixsupport_win32.c
-@@ -251,7 +251,7 @@ void caml_win32_maperr(DWORD errcode)
+@@ -259,7 +259,7 @@ void caml_win32_maperr(DWORD errcode)
  #undef EACCESS
  #define EACCESS EACCES
  
