--- a	2023-09-28 18:46:55.039260811 +0100
+++ b	2023-09-28 18:46:55.055261404 +0100
@@ -15,7 +15,7 @@
 diff --git a/stdlib/filename.ml b/stdlib/filename.ml
 --- a/stdlib/filename.ml
 +++ b/stdlib/filename.ml
-@@ -347,7 +347,7 @@ let temp_file ?(temp_dir = Domain.DLS.get current_temp_dir_name) prefix suffix =
+@@ -367,7 +367,7 @@ let temp_file ?(temp_dir = Domain.DLS.get current_temp_dir_name) prefix suffix =
        close_desc(open_desc name [Open_wronly; Open_creat; Open_excl] 0o600);
        name
      with Sys_error _ as e ->
@@ -24,7 +24,7 @@
    in try_name 0
  
  let open_temp_file ?(mode = [Open_text]) ?(perms = 0o600)
-@@ -359,5 +359,16 @@ let open_temp_file ?(mode = [Open_text]) ?(perms = 0o600)
+@@ -379,5 +379,16 @@ let open_temp_file ?(mode = [Open_text]) ?(perms = 0o600)
        (name,
         open_out_gen (Open_wronly::Open_creat::Open_excl::mode) perms name)
      with Sys_error _ as e ->
