--- a	2023-09-28 18:46:56.335308820 +0100
+++ b	2023-09-28 18:46:56.343309116 +0100
@@ -1,7 +1,7 @@
 diff --git a/Makefile b/Makefile
 --- a/Makefile
 +++ b/Makefile
-@@ -1856,6 +1856,11 @@ endif
+@@ -1859,6 +1859,11 @@ endif
  
  include .depend
  
@@ -28,7 +28,7 @@
 diff --git a/configure.ac b/configure.ac
 --- a/configure.ac
 +++ b/configure.ac
-@@ -213,6 +213,8 @@ AC_SUBST([compute_deps])
+@@ -215,6 +215,8 @@ AC_SUBST([naked_pointers_checker])
  AC_SUBST([ocaml_bindir])
  AC_SUBST([ocaml_libdir])
  AC_SUBST([QS])
