--- a	2023-09-28 18:48:25.026522140 +0100
+++ b	2023-09-28 18:48:25.054523133 +0100
@@ -50,7 +50,7 @@
 diff --git a/asmcomp/cmmgen.ml b/asmcomp/cmmgen.ml
 --- a/asmcomp/cmmgen.ml
 +++ b/asmcomp/cmmgen.ml
-@@ -814,7 +814,7 @@ and transl_ccall env prim args dbg =
+@@ -821,7 +821,7 @@ and transl_ccall env prim args dbg =
            | Pint32 -> XInt32
            | Pint64 -> XInt64 in
          (xty, transl_unbox_int dbg env bi arg)
@@ -59,7 +59,7 @@
          (XInt, untag_int (transl env arg) dbg)
    in
    let rec transl_args native_repr_args args =
-@@ -835,7 +835,7 @@ and transl_ccall env prim args dbg =
+@@ -842,7 +842,7 @@ and transl_ccall env prim args dbg =
      | Same_as_ocaml_repr -> (typ_val, fun x -> x)
      | Unboxed_float -> (typ_float, box_float dbg)
      | Unboxed_integer bi -> (typ_int, box_int dbg bi)
